<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 36</title>
</head>

<body>

<div id="maintitle"><h1>Convert Marker QF to fewi</h1>
    <ul>
        <li>User Story 36 - Convert Marker QF from javawi to fewi</li>
    </ul>
</div><div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
	<ol>
	   <li>Re-implement the Marker QF in the fewi product (from java wi).  The Marker Summary is not part of this work.
	   <ol><li>Headings, links, intro text
	       <ul><li>heading = Genes and Markers Query Form</li>
	           <li>user help link from ? - GENE_help.shtml</li>
	           <li>text and anchor tag to output options ribbon</li>
	           <li>Search and Reset buttons - top and bottom of page</li>
	           
	       </ul></li>
	       <li>Gene/Marker ribbon
	       <ol><li>Gene/Marker symbol/Name field (linked to User Help)
	           <ol><li>operators
	               <ul><li>contains (default)</li>
	                   <li>equals</li>
	                   <li>begins</li>
	               </ul></li>
	               <li>Input box
	               <ul><li>alpha string(s) valid -- ids are not accepted</li>
	                   <li>comma-delimited lists accepted</li>
	                   <li>quotation marks used for exact matching</li>
	               </ul></li>
	               <li>Search scope
	               <ul><li>current & old symbols/names, synonyms, alleles, homologs (default)
	                      <ul><li>markers = marker names, symbols, synonyms</li>
	                          <li>alleles = allele names</li>
	                          <li>homologs = non-mouse symbols, names & human/rat synonyms</li>
	                      </ul></li>
	                   
	                   <li>current symbols/names (markers only)</li>
	               </ul></li>
	           </ol></li>
	            
	           
	       </ol></li>
	       <li>Feature Type ribbon
	       <ol><li>Display the hierarchical list of Feature Types with counts</li>
	           <li>Default - none checked which is the same as all checked</li>
	           <li>check boxes - clicking a parent term also checks children terms</li>
	           <li>minus signs - indicates discrete items at a lower level have been selected</li>
	           <li>Show/Hide - Show: display all; Hide: display only single highest term; Default: custom, see Public </li>
	           <li>Reset checks button - blanks out all boxes</li>
	           <li>Tool tip with term definition - displays with mouse over; includes close option; displays for n secs</li>
	           <li>Text blurb and link to User Help - included for javascript and non-javascript versions</li>
	           <li>Provide a display of this table for javascript disabled browsers.</li>
	       </ol></li>
	       <li>Map position ribbon
	       <ol><li>Genome Coordinates and cM position cannot be combined in one search</li>
	           <li>Chromosome field (linked to User Help)
	           <ol><li>Display multi-select box with all possible chromosomes (Any, 1-19, X, Y, XY, UN, MT)</li>
	               <li>Default = Any</li>
	           </ol></li>
	           <li>Genome Coordinates field (linked to User Help)
	           <ol><li>Operator: between (remove pull-down with other choices)</li>
	               <li>input box for coordinates</li>
	               <li>formats for coordinates - hyphen or AND (e.g. nnn-nnn; nnn AND nnn)
	               
	               <li>pull down with bp or Mbp - default is bp</li>
	               <li>display hints</li>
	           </ol></li>
	           <li>cM Position field (linked to User Help)
	           <ol><li>Operator: between (remove pull-down with other choices)</li>
	               <li>location on page - since we are removing Cytoband move cM up next to Coordinate field</li>
	               <li>input box for cM values - hyphen or AND (e.g. nn-nn; nn AND nn)</li>
	               <li>display hints</li>
	           </ol></li>
	           <li>Cytogenetic Band - Remove</li>
	           <li>Marker Range field (linked to User Help)
	           <ol><li>Operator: between</li>
	               <li>2 input boxes - only current markers with coordinates</li>
	               <li>QTLs not allowed</li>
	               <li>Markers must be on same chromosome</li>
	               <li>display hint</li>
	           </ol></li>
	       </ol></li>
	       <li>Gene Ontology classifications ribbon
	       <ol><li>Gene Ontology (GO) Classifications field (linked to User Help) 
	           <ol><li>Operator: contains (remove pull-down with other choices)</li>
	               <li>Input box for terms or IDs</li>
	               <li>AND valid for either terms or IDs -- markers returned must match both</li>
	           </ol></li>
	           <li>Ontology check boxes
	           <ol><li>List 3 GO Ontologies with check boxes</li>
	               <li>Default - all boxes checked</li>
	           </ol></li>
	           <li>Link to GO Browser</li>
	       </ol></li>
	       <li>Protein domains ribbon
	       <ol><li>InterPro Protein Domains field (linked to User Help)
	           <ol><li>Input box for terms or IDs</li>
	               <li>Operator - this is a contains search -- change to display "contains" to match GO search.</li>
	           </ol></li>
	           <li>Link to InterPro Public Report</li>
	       </ol></li>
	       <li>Mouse phenotypes & mouse models of human disease ribbon
	       <ol><li>Phenotype/Human Disease box (linked to User Help)
	           <ul><li>search MP and OMIM terms and IDs</li>
	               <li>AND and OR recognized</li>
	               <li>quotes recognized</li>
	               <li>partial word matching (contains?)</li>
	           </ul></li>
	           <li>Pop-up high-level System box
	           <ul><li>Multi-select box with the MP high level terms</li>
	               <li>Enter custom text into Pheno box for each selected term</li>
	               <li>Pop-up - new window and small</li>
	               <li>Text heading</li>
	               <li>Help link - ? - ALLELE_phenotype_highlevel_help.shtml (not currently using the FAQ style pop-up for this help page - change?)</li>
	               <li>Add to Query button - adds text and closes pop-up</li>
	               <li>Cancel button - closes pop-up</li>
	           </ul></li>
	           <li>Link to MP Browser</li>
	           <li>Link to OMIM Browser</li>
	       </ol></li>
	       <li>Clone Collection - Remove</li>
	       <li>Sorting and output format ribbon
	       <ol><li>Sort by field (linked to User Help)
	           <ol><li>Nomenclature - default</li>
	               <li>Genome Coordinates</li>
	               <li>cM Position - remove</li>
	           </ol></li>
	           <li>Maximum returned field (linked to User Help)
	           <ol><li>500 - default</li>
	               <li>Input box to specify a custom return - remove</li>
	               <li>no limit checkbox</li>
	           </ol></li>
	           <li>Output field (linked to User Help)
	           <ol><li>Web - default</li>
	               <li>Excel</li>
	               <li>Tab-delimited</li>
	               <li>Tab-delimited to ftp site - remove</li>
	           </ol></li>
	       </ol></li>
	   </ol></li>
	   <li>Update links to Marker Query Form in MGI
	   <ol><li>toolbar pull-down (webshare)</li>
	       <li>All Search menu (python wi)</li>
	       <li>Marker Summary (Modify Search link) (python wi)</li>
	       <li>Submission CGIs (mgi home)</li>
	       <li>Mini homes, User Help, FAQ</li>
	   </ol></li>
	   <li>Forward old URL to new URL</li>
	   <li>Display correctly in all supported browsers (IE, Firefox, Safari, Chrome)</li>
	   <li>Verify "You searched for" display on Marker Summary has not changed</li>
	   <li>Verify Marker Summary results are the same</li>
	   
	   
	</ol>
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
    		
    		<li>Verify headings, links, buttons at top of page and buttons at bottom of page
    		<p concordion:assertTrue="false">FAIL - jlewis 9/4/2013 - Browser back button clears some fields (released with this issue - low priority)</p>  
    		<ol>
    		   <li>Verify visual display - all browsers - Safari/MAC, Firefox/MAC, IE/PC, Firefox/PC</li>
    		   <li>Verify buttons, links - all browsers</li>
    		   
    	    </ol></li>
    	    
    	   <li>Verify the Gene/Marker Ribbon
    		<p concordion:assertTrue="true">PASS - jlewis 9/4/2013</p>    
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Search marker nomenclature text: CONTAINS <i>pax</i> in CURRENT & OLD...  Expect: 58 results.</li>
    		   <li>Search allele name text: BEGINS <i>yellow</i> in CURRENT & OLD...  Expect: 4 results.</li>
    		   <li>Search homolog nomenclature: EQUALS <i>Dlx5</i> in CURRENT & OLD...  Expect: 2 results.</li>
    		   <li>Search marker nomenclature:  CONTAINS <i>Trp53</i> in CURRENT SYMBOLS/NAMES.  Expect:  28 results</li>
    		   
    	    </ol></li>
    	  
    	  <li>Verify the Feature Type Ribbon - javascript enabled
    		<p concordion:assertTrue="true">PASS - jlewis 9/4/2013</p>    
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Search using the default (all feature types valid for return).</li>
    		   <li>Search after selecting all boxes.  Expect same results as if checking no boxes.</li>
    		   <li>Search selecting just one child feature type.  Expect only markers of that feature type returned</li>
    		   <li>Search selecting a parent feature type.  Expect markers with feature types that match parent and also children to be returned.</li>
    		   <li>Search selecting multiple child feature types.  Expect only those specific types to be returned.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Verify the Feature Type Ribbon - javascript disabled
    		<p concordion:assertTrue="true">PASS - jlewis 9/9/2013 - link to User Help broken</p>    
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Search using the default (all feature types valid for return).</li>
    		   <li>Search after selecting all boxes.  Expect same results as if checking no boxes.</li>
    		   <li>Search selecting just one child feature type.  Expect only markers of that feature type returned</li>
    		   <li>Search selecting a parent feature type.  Expect markers with feature types that match parent and also children to be returned. (I'm not sure how this works when javascript disabled)</li>
    		   <li>Search selecting multiple child feature types.  Expect only those specific types to be returned.</li>
    		   
    		   
    	    </ol></li>
    	    
    	    <li>Verify the Map Position Ribbon
    		<p concordion:assertTrue="true">PASS - jlewis 9/4/2013</p>    
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Search by Chromosome MT only.  Expect 42 results.</li>
    		   <li>Search for multiple chromosomes: nn and nn.  Expect nn results. (broken on public - entered a TR)</li>
    		   <li>Search by Chr and Coordinates: chr=5 and coords=28440000-28470000 bp.  Expect 4 results.</li>
    		   <li>Search by Chr and Coordinates: chr=5 and coords=28440000 AND 28470000 bp.  Expect 4 results.</li>
    		   <li>Search by Chr and Coordinates: chr=2 and Coords=125.0-126.0 Mbp.  Expect 33 results.</li>
    		   <li>Search by Chr and cM:  chr=6 and between 12.2-15.5.  Expect 162 results.</li>
    		   <li>Search by Chr and cM:  chr=6 and between 12.2 and 15.5.  Expect 162 results. (fyi-Public WI requires AND be in lower case)</li> 
    		   <li>Search by Marker range:  between Vat1 and Nbr1.  Expect 14 results.</li>
    		   <li>Search by Chr, Coordinates, and cM.  Expect error message - not a valid search.</li>
    		</ol></li>
    		

    	    <li>Verify the GO classifications Ribbon
    		<p concordion:assertTrue="true">PASS - jlewis 9/4/2013</p>    
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Search for a single GO ID, GO:0030375 -- all ontologies.  Expect 5 markers associated to GO id.</li>
    		   <li>Search for a single GO ID, GO:0030375 -- deselect Molecular Function ontology.  Expect 0 results.</li>
    		   <li>Search for multiple GO IDs, GO:0030375 and GO:0030518 -- all ontologies.  Expect 3 results.</li>
    		   <li>Search for multiple GO IDs, GO:0030375 and GO:0030518 and GO:0016922 -- all ontologies.  Expect 2 results.</li>
    		   <li>Search for a single token, deaminase -- all ontologies.  Expect 31 results.</li>
    		   <li>Search for multiple tokens, <i>AMP deaminase activity</i> -- all ontologies.  Expect 3 results.</li>
    		   <li>Search for multiple terms using the AND operator, <i>cadherin binding AND Kit signaling pathway</i> -- all ontologies.  Expect 1 result.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Verify the Protein domains Ribbon
    		<p concordion:assertTrue="true">PASS - jlewis 9/4/2013</p>   
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Search for a single InterPro ID, IPR000007.  Expect 5 results.</li>
    		   <li>Search for an invalid InterPro ID, IPR0000xx.  Expect 0 results.</li>
    		   <li>Search for multiple InterPro IDs IPR000007 AND IPR005398.  Expect 3 results.</li>
    		   <li>Search for a single token, <i>tubby</i>.  Expect 6 results.</li>
    		   <li>Search for multiple tokens, <i>tubby AND scramblase</i>.  Expect 1 result.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Verify the Phenotype Ribbon
    		<p concordion:assertTrue="true">PASS - jlewis 9/9/2013</p>  
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Search for a single MP id, MP:0000588. Expect 21 results.</li>
    		   <li>Search for a single MP id, MP:0004412.  Expect 14 results.</li>
    		   <li>Search for multiple MP ids - using AND: MP:0000588 AND MP:000007. Expect 8 results.</li>
    		   <li>Search for multiple MP ids - using OR:   MP:0000588 OR MP:0004412.  Expect 35 results.</li>
    		   <li>Search for a single OMIM id, 125800.  Expect 7 results.</li>
    		   <li>Search for a single OMIM id, 613282.  Expect 4 results.</li>
    		   <li>Search for multiple OMIM ids - AND: 125800 AND 613282.  Expect 1 result. -- Doesn't work on Public (no results returned)</li>
    		   <li>Search for multiple OMIM ids - OR: 125800 OR 613282.  Expect 10 results.</li>
    		   <li>Search for a single text token, <i>albino</i>.  Expect 82 results.</li>
    		   <li>Search for a single text token, <i>alzheimer</i>.  Expect 98 results.</li>
    		   <li>Search for multiple text tokens, <i>Alstrom Syndrome</i>.  Expect 1 result.</li>
    		   <li>Search for multiple text tokens, <i>Curly whiskers</i>.  Expect 95 results.</li>
    		   <li>Search for multiple text tokens, <i>curly AND whiskers</i>.  Expect 95 results.</li>
    		   <li>Search for multiple text tokens, <i>curly OR whiskers</i>.  Expect 337 results.</li>
    		   <li>Search using quotes, "curly whiskers".  Expect 80 results.</li>
    		   <li>Search using quotes, "Curly whiskers" AND "eyelids open at birth" .  Expect 15 results.</li>
    		   <li>Search using quotes, "Curly whiskers" OR "eyelids open at birth" .  Expect 181 results.</li>
    		   <li>Search with text entered using the High-level pheno pop-up.  Select 1 high level term, <i>Other Phenotype</i>. Expect 442 results.</li>
    		   <li>Search with text entered using the High-level pheno pop-up.  Select 2 high level terms, <i>Craniofacial</i> and <i>Pigmentation</i>. Expect 182 results</li>
    		   <li>Search with text entered manually, <i>albino</i>, and using the pheno pop-up, term <i>Vision/Eye</i>.  Expect 37 results.</li>
    		   <li>Select every term on the pop-up to verify the text generated matches Public.</li> 
    		   
    	    </ol></li>
    	    
    	    <li>Verify the Sorting and Output Format Ribbon
    		<p concordion:assertTrue="true">PASS - jlewis 9/4/2013</p>    
    		<ol>
    		   <li>Verify visual display - all browsers</li>
    		   <li>Verify buttons, links, defaults, options - all browsers</li>
    		   <li>Verify Sort by default of Nomenclature.  Expect results sorted by matching text</li>
    		   <li>Modify Sort by to Genome Coordinates.  Expect results sorted by Chr/Coords.</li>
    		   <li>Verify Maximum returned default of 500.  Expect limit of 500 results.</li>
    		   <li>Modify Maximum returned to no limit.  Expect all search results returned.</li>
    		   <li>Verify Output default of Web.  Expect results in Web format.</li>
    		   <li>Modify Output to Excel.  Expect results in Excel format.</li>
    		   <li>Modify Output to Tab-delimited.  Expect results in Tab format.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Verify links to Marker QF
    	    <p concordion:assertTrue="true">PASS - jlewis 9/9/2013</p>    
    		<ol>
    		   <li>Check toolbar, menus, Marker Summary, Submission CGIs, Minihomes, User Help, FAQ</li>
    		</ol></li>
    	    
		</ol>
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>