<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 15</title>
</head>

<body>

<div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Acceptance Criteria</a> | <a href="#manual">Manual Tests</a> | <a href="#auto">Automated Tests</a></p>
</div>

<div id="maintitle"><h1>Assign Coordinates to Markers w/no Gene Model</h1>
    <ul>
        <li>User Story 15, Support bulk loading to assign coordinates directly to markers that do not have gene models.</li>
    </ul>
</div>

<div id="acceptance">
    <h2>Test Requirements</h2>
	<ul>
		<li>Assign coordinates to any Marker Type.
		    <ul><li>types include: Gene, Pseudogene, DNA segment, Cyto Marker, QTL, BAC/YAC, Other Genome Feature, Complex/Cluster/Region</li>
		    </ul>
		<li>Load coordinates for tRNA genes</li>
		<li>Load coordinates for multiple markers in one input file</li>
		<li>Allow curator to specify Genome Build to display in the Prod WI in the Curator input file</li>
		<li>Allow curator to specify Provider text  by marker to display in the Prod WI in the Curator input file</li>
		<li>Allow curator to override previously assigned coordinates in the Curator input file by marker</li>
		<li>Allow curator to remove previously assigned coordinates by marker</li>
	</ul>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		<ol>
    		<li>Loaded chr, coordinates, provider text, and build text are displayed in the Sequence Map section of the Production WI Marker Detail page for all marker types.  
    		<p concordion:assertTrue="true">jlewis 7/13/12</p> 
    		<ol>
    			<li>Display  marker <b>1700016D08Rik</b>.  This marker has a marker type of <b>Gene</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr16:3908825-3909192 bp, -  strand <br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>Pou5f1-rs8</b>.  This marker has a marker type of <b>Pseudogene</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chrX:50232039-50233572 bp, + strand <br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>D12Mit100</b>.  This marker has a marker type of <b>DNA Segment</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr12:105761460-105761553  bp, + strand <br>(From UniSTS  annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>Del(5D5Mit73-D5Mit351)1J</b>  This marker has a marker type of <b>Cyto Marker</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr5:26927527-37410359 bp<br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>Arbq2</b>.  This marker has a marker type of <b>QTL</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chrY:484252-484471 bp<br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>3-209M19-S</b>.  This marker has a marker type of <b>BAC/YAC</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr2:103286918-103286918 bp <br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>H3093D10</b>.  This marker has a marker type of <b>Other Genome Feature</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr9:65557564-65558161 bp, - strand<br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>Snord116</b>.  This marker has a marker type of <b>Complex/Cluster/Region</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr7:66880887-67119801 bp<br>(From MGI annotation of NCBI Build XX)</b></b></li></ul>
    			
    			<li>Display  marker <b>Tg(Ins1-Tag,-luc)1Gcr</b>.  This marker has a marker type of <b>Transgene</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr3:156098318-156098318 bp<br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
       		</ol>
       		
    		<li>Loaded chr, coordinates, provider text, and build text are displayed in the Sequence Map section of the Production WI Marker Detail page for tRNA genes.  
    		<p concordion:assertTrue="true">jlewis 7/13/12</p>  
    		<ol>
    		<li>Display  marker <b>n-Ta1</b>.  This marker has a marker type of <b>Gene</b> and feature type of <b>tRNA gene</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chr11:51790041-51790112 bp, + strand <br>(From GtRNAdb annotation of NCBI Build XX)</b></li></ul>
    		</ol>
    		<li>Verify newly assigned coordinates override previously assigned coordinates from the <i>same</i> collection.<ol>
    		<p concordion:assertTrue="true">7/13/12</p> 	
    			<li>Collection = QTL (Marker Type = QTL)</li>
    				<ul><li>Display marker <b>Ity2a</b>.
    				<ul><li>Verify the Sequence Map display: <br><b>chr11:46268437-46294757 bp<br>genome coordinates based on the boundaries of the QTL region <br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			    </ul>
    			<li>Collection = NCBI uniSTS (Marker Type = DNA Segment)</li>
    			    <ul><li>Display marker <b>D12Mit100</b>
    			    <ul><li>Verify the Sequence Map display: <br><b>chr12:105761460-105761553 bp<br>(From UniSTS annotation of NCBI Build XX)</b></li></ul>
    			    </ul>
    		</ol>
    		
    		<li>Verify newly assigned coordinates override previously assigned coordinates from a <i>different</i> collection.<ol>
    		<p concordion:assertTrue="true">jlewis 7/13/12</p> 	
    			<li>Collection = roopenian STS (Marker Type = DNA Segment)</li>
    			    <ul><li>Display marker <b>D1Dcr15</b>
    			    <ul><li>Verify the Sequence Map display: <br><b>chr1:162387261-162387360 bp<br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			    </ul>
    			<li>Collection = TR9601 DNA sequences (Marker Type = DNA Segment)</li>
    			    <ul><li>Display marker <b>30.MMHAP16FRB12</b>
    			    <ul><li>Verify the Sequence Map display: <br><b>chrX:45460000-45460183 bp<br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			    </ul>
    		</ol>
    		
    		<li>Markers from a new collection specified in the file display assigned coordinates, provider, and build text.
    		<p concordion:assertTrue="true">jlewis 7/13/12</p> 
    			<ol><li>Display marker <b>Snord116l3</b>.  This marker is part of a new collection, <b>MGI Curation</b>.</li></ol>
    			<ul><li>Verify the Sequence Map display: <br><b>Chr7:67024347-67024436 bp, -  strand <br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    		
    		
    		<li>Verify assignment of coordinates with special cases in re: to chromosomes<ol>
    		<p concordion:assertTrue="true">jlewis 7/13/12</p> 
    		    <li>Display marker <b>mt-Olr</b>. This marker is on the <b>MT chromosome</b>.</li>
    		    <ul><li>Verify the Sequence Map display: <br><b>chrMT:4952-5327 bp, + strand<br>(From MGI annotation of NCBI Build XX)</b></li></ul>
    			
    		</ol>
    		
    		<li>Markers with no coordinate assignment via the load (or gene models) don't display coordinates or credit/build text. (as-of July 9th - no coordinates on Public WI for the following markers)  The results are the same for each of these - <b>no display of coordinates, provider, or build</b><ol>
    		<p concordion:assertTrue="true">jlewis 7/13/12</p> 
    			<li>Display marker <b>Gm4146</b>. This marker has a marker type of <b>Gene</b>.</li>
                <li>Display marker <b>412H6</b>.  This marker has a marker type of <b>DNA segment</b></li>
                <li>Display marker <b>Del(18)20H</b>.  This marker has a marker type of <b>Cyto Marker</b>.</li>
                <li>Display marker <b>Aasm3</b>.  This marker has a marker type of <b>QTL</b>.</li>
                <li>Display marker <b>Gm10270</b>. This marker has a marker type of <b>Pseudogene</b>.</li>
                <li>Display marker <b>51B23</b>. This marker has a marker type of <b>BAC/YAC</b>.</li>
                <li>Display marker <b>Ap3s1-rs3</b>.  This marker has a marker type of <b>Other Genome Feature</b>.</li>
                <li>Display marker <b>Mirc18</b>.  This marker has a marker type of <b>Complex/Cluster/Region</b>.</li>
                <li>Display marker <b>Tn(sb-Tyr)1799B.CA1Ove</b>.  This marker has a marker type of <b>Transgene</b>.</li>
    		</ol>
    		
    		<li>Verify previously assigned coordinates are deleted
    		<p concordion:assertTrue="false">jlewis 7/13/12</p> 
    		<ol>
    			<li>Display marker <b>Mir24-1</b>. Marker type is <b>Gene</b> <b>TEST FAILED</b>
    			    <ul><li>Verify the Sequence Map display: <b>no coordinates, provider, or build text displayed</b></li></ul>
    			<li>Display marker <b>Aaiq2</b>.  Marker type is <b>QTL</b>.
    			    <ul><li>Verify the Sequence Map display: <b>no coordinates, provider, or build text displayed</b></li></ul>
    		</ol>
    	    <li>Verify that markers from collection TR9601 now display the updated display text. 
    		<p concordion:assertTrue="true">jlewis 7/13/12</p> 
    		<ol>
    			<li>Display marker <b>D0Nds1</b>. Marker type is <b>DNA Segment</b>
    			    <ul><li>Verify the Sequence Map display: <br><b>chr15:76853482-76853646 bp, - strand<br>(From MGI annotation of NCBI Build 37)</b></li></ul>
    			</ol>
    			
    			
		</ol>
	</div>
	<div id="auto">

 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>