<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 39</title>
</head>

<body>

<div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#requirements">Test Requirements</a> 
	| <a href="#manual">Manual Tests</a> | <a href="#auto">Automated Tests</a></p>
</div>

<div id="maintitle">
<h1>Public Reports contain correct coords for markers with directly associated coordinates</h1>
    <ul>
        <li>User Story 39, Public Reports contain correct coords for markers 
        with directly associated coordinates.</li>
        <li>Reports included in test: MGI_MRK_coord.rpt (new), MGI_AllGenes.rpt, 
        MRK_list1.rpt, MRK_list2.rpt, MGI_Coordinate.rpt, MGI_EntrezGene.rpt, 
        MGI_GTDNA.rpt, MGI_GTGUP.rpt, MGI_GTRNA.rpt, MGI_GTdbGSS_info.rpt, 
        MGI_IKMC_DNA_GT.rpt, MGI_IKMC_DNA_GT_info.rpt, MGI_iphone_app.rpt, 
        MRK_ENSEMBL.rpt, MRK_List.rpt, MRK_Sequence.py, MRK_VEGA.rpt, 
        Affy_1.0_ST_mgi.rpt, Affy_430_2.0_mgi.rpt, Affy_U74_mgi.rpt
		</li>
    </ul>
</div>

<div id="requirements">
    <h2>Test Requirements</h2>
	<ul>
		<li>Reports show curator assigned coordinates</li>
		<li>Reports show gene model coordinates when no curator assigned coordinates exist</li>
		<li>Reports that display provider data show curator assigned provider</li>
		<li>MRK_coordinate report is split into 2 new reports; MGI_gene_model_coord.rpt 
		and MGI_MRK_coord.rpt</li>
		<li>New reports created for tRNA markers, DNA segment markers, and heritable phenotypic markers</li>
	</ul>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>

		<ol>
			<li>Confirm new report MGI_MRK_coord.rpt is created
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Report title is MGI Marker coordinates</li>
				<li>Report appears in the Mouse Genetic Markers section of the 
				reports index page</li>
				<li>Report contains following columns:  MGI ID, Marker Type, 
				Feature Type, Symbol, Name, Chr, Start, End, Strand, Provider, 
				Display</li>
				<li>Markers with no coordinates that has curator assigned coordinates removed are not included in report.
				<ul>
					<li>Verify that marker <b>XX</b> does not appear in report</li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>1110002O04Rik</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Mir1193</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
	
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>			
			<li>Confirm that markers have the correct coordinates in 
			MGI_AllGenes.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Ity2a</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Mir191</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>				
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in MRK_list1.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>D12Mit100</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Qk</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>				
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in MRK_list2.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>D1Dcr15</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Tnxa</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>								
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>			
			<li>Confirm changes to MGI_Coordinate.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Report is replaced by MGI_gene_model_report.rpt</li>
				<ul>
					<li>Report title is MGI Gene Model Coordinates</li>
					<li>Report contains following columns: MGI accession id, 
					marker type, marker symbol, marker name, 
					representative genome id, representative genome chromosome, 
					representative genome start, representative genome end, 
					representative genome strand, representative genome build,	
					Entrez gene id, NCBI gene chromosome, NCBI gene start, 
					NCBI gene end, NCBI gene strand, Ensembl gene id, 
					Ensembl gene chromosome, Ensembl gene start, Ensembl gene end, 
					Ensembl gene strand, VEGA gene id, VEGA gene chromosome, 
					VEGA gene start, VEGA gene end, VEGA gene strand</li>
				</ul>
			</ol>
			</li>
			<li>Confirm changes to MGI_QTLAllele.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>				
				<li>Report now contains following new columns: chr, start,
				end</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Arbq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MGI_IKMC_DNA_GT_info.rpt 
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Qk</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>					
			<li>Confirm that markers have the correct coordinates in 
			MGI_EntrezGene.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>mt-Nd6</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>			
			<li>Confirm that markers have the correct coordinates in 
			MGI_GTDNA.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>D12Mit100</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Pde1b</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MGI_GTGUP.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>XD1Dcr15</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Mir191</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MGI_GTRNA.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Mir199s</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Qk</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MGI_GTdbGSS_info.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>1110002O04Rik</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Tnxa</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MGI_IKMC_DNA_GT.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Ity2a</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Zpf53</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MGI_IKMC_DNA_GT_info.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>D12Mit100</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Cldn20</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MGI_iphone_app.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>D1Dcr15</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>mt-Nd6</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MRK_ENSEMBL.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with Ensembl gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Mir1193</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with Ensembl gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MRK_Sequence.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Mir199s</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Pde1b</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Aaiq2</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>Mir24-1</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			MRK_VEGA.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with VEGA gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Tnxa</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with VEGA gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			Affy_1.0_ST_mgi.rpt
			<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Qk</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in  Affy_430_2.0_mgi.rpt<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Qk</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>
			<li>Confirm that markers have the correct coordinates in 
			 Affy_U74_mgi.rpt<p concordion:assertTrue="false">new test</p>
			<ol>
				<li>Coordinates are reported properly for markers with no gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have newly loaded coordinates.<br/>
				Find marker <b>Qk</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with no gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
				<li>Coordinates are reported properly for markers with gene 
				model that have deleted curator assigned coordinates.<br/>
				Find marker <b>XX</b>. Verify the location fields:
				<ul>
					<li>chromosome: <b>x</b></li>
			  		<li>cM position: <b>x</b></li>
					<li>start coordinate: <b>x</b></li>
					<li>end coordinate: <b>x</b></li>
					<li>strand: <b>x</b></li>
				</ul>
				</li>
			</ol>
			</li>													
		</ol>
	</div>
	<div id="auto">

 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>
