<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 31</title>
</head>

<body>

<div id="maintitle"><h1>Convert PIRSF Page to fewi</h1>
    <ul>
        <li>User Story 31 - Convert PIRSF page & implement n-m data change</li>
    </ul>
</div><div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
	<ol>
	    <li>Page heading area (above the table of PIRSF mouse members)
	    <ul>
	       <li>Page heading = Protein Superfamily Detail</li>
	       <li>User help "?" link = userhelp/pirsf_detail_help.shtml (same as on Public now)</li>
	       <li>PIRSF Classification ribbon: display PIRSF term and ID; link ID to PIRSF site</li>
	       <li>text blurb - see mockup for details</li>
	       <li>MGI Superfamily Information - button that triggers a pop-up with text - see mockup for text.  Same style as Homology Class page.</li>
	       <li>Download/Blast button controls - use standard found elsewhere</li>
	       <li>Add mouse-over tag to the Counts column (just like found on Disease Detail page)
	       <ul><li>"The number of Mouse, Human, and Rat genes present in a corresponding Homology Class are indicated."
	       </ul></li>
	    </ul></li>
	    <li>Table of PIRSF mouse members
	    <ul>
	      <li>PIRSF mouse members are determined via an association load of PIRSF ids to MGI ids.</li>
	      <li>PIRSF mouse members may be part of a HomoloGene Class or not.</li>
	      <li>Display table of PIRSF mouse members
	      <ol>
	         <li>Mouse PIRSF Members
	            <ul><li>List markers in alpha order (dumb order okay, but smart preferred if easy to do)</li>
	                <li>Link marker symbol to Marker Detail page</li>
	            </ul></li>
	         
	         <li>Chr - the front end database is different than MGD -- chromosome is stored in several places.  Use the following hierarchy to find the chromosome
	            <ul><li>Chromosome stored with coordinates, if it exists</li>
	                <li>Chromosome stored with cM, if it exists</li>
	                <li>Chromosome stored with Cytoband, if it exists</li>
	                <li>then what?  (Jon?)
	            </ul> </li>
	         <li>Homology Class
	            <ul><li>If the mouse marker in the PIRSF class is also in a Homlogy class - display the class ID</li>
	                <li>If the mouse marker in the PIRSF class is NOT in a Homology class - display "none"</li>
	                <li>Link ID to the MGI Homology Class page</li>           
	            </ul></li>
	         <li>Homology Class, Mouse : Human : Rat
	            <ul><li>list the number of markers in the Homology class for mouse, human, and rat (no other organisms).</li>
	                <li>display 0 if there are none for a particular organism.</li>
	                <li>when the PIRSF mouse member is not in a Homology Class -- leave this field blank</li>
	            </ul></li>
	         <li>Select button for dowload/forward
	         <ul><li>multiple members can be selected for download</li>
	             <li>selecting a mouse member results in REPRESENTATIVE proteins being downloaded for mouse, human, and rat</li>
	             <li>protein sequences are downloaded for all members of the Homology class - including other mouse markers</li>
	             <li>if there are no representative proteins for a particular row of the table -- don't display the checkbox (yes, this cases exists)</li>
	         </ul></li>
	      </ol>
	    </ul>
	  
	   
	</ol>
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
    		
    		<li>Verify page heading area: text, pop-ups, links
    		<p concordion:assertTrue="true">PASS - jlewis 8/15/2013</p>  
    		<ol>
    		   <li>Display PIRSF038172.  Verify links, pop-ups, and text using mockup.</li>
    	    </ol></li>
    	    
    	    <li>Verify PIRSF Members and Homology Class columns - display and links
    		<p concordion:assertTrue="true">PASS - jlewis 8/15/2013</p>  
    		<ol>
    		   <li>Display PIRSF page with only 1 member with a homology class, PIRSF000546.</li>
    		   <li>Display PIRSF page with only 1 member that is not in a homology class, PIRSF002466.</li>
    		   <li>Display PIRSF page with multiple members all with a homology class, PIRSF038195.</li>
    		   <li>Display PIRSF page with multiple members and at least one without a homology class, PIRSF023714</li>
    		   <li>Display PIRSF page to verify smart alpha sort, PIRSF001015.</li>
    	    </ol></li>
    	    
    	    <li>Verify Chromosome column
    		<p concordion:assertTrue="true">PASS - jlewis 8/15/2013</p>   
    		<ol>
    		   <li>Display PIRSF page with a member that has UN chromosome, PIRSF001015</li>
    		   <li>Display PIRSF page with a member that has coordinates, PIRSF009400</li>
    		   <li>Display PIRSF page with a member that has no coordinates, PIRSF016410 and PIRSF005608</li>
    		   <li>Display PIRSF page with a member that has no coordinates and no cM position, but has a cytoband position, PIRSF037964</li>
    		   <li>Display PIRSF page with a member that is on MT chromosome, PIRSF001283</li>
    	    </ol></li>
    	    
    	    <li>Verify Homology Class counts column
    		<p concordion:assertTrue="true">PASS - jlewis 8/15/2013</p>   
    		<ol>
    		   <li>Display a PIRSF page with a member that has 1 marker in each organism, PIRSF001734.</li>
    		   <li>Display a PIRSF page with a member that has multiple non-mouse markers in its homology class, PIRSF001635.</li>
    		   <li>Display a PIRSF page with a member that has multiple mouse markers in its homology class, PIRSF005491.</li>
    		   <li>Display a PIRSF page with a member that has 0 human markers in its homology class, PIRSF006060.</li>
    		   <li>Display a PIRSF page with a member that has 0 rat markers in its homology class, PIRSF037272.</li>
    		   <li>Display a PIRSF page with a member that has only mouse markers in its homology class, PIRSF001015.</li>
    		   <li>Display a PIRSF page with a member that has no homology class and therefore no counts, PIRSF003110.</li>
    	    </ol></li>
    	    
    	    <li>Verify download of protein sequences
    		<p concordion:assertTrue="true">PASS - jlewis 8/15/2013</p> 
    		<ol>
    		   <li>Verify the select all, invert, reset, and GO buttons.</li>
    		   <li>Download a single row for a family with more than one member and one that has sequences for all 3 organisms, PIRSF001199.</li>
    		   <li>Download 2 rows of a family page with more than 2 rows, PIRSF001873.</li>
    		   <li>Download all rows for a family with 3 or more members, PIRSF002051.</li>
    		   <li>Download all rows for a family with just 1 member, PIRSF002089.</li>
    		   <li>Download all rows for a family with a member that has no select box, PIRSF005491.</li>
    		   <li>Download an Ensembl protein sequence, PIRSF006060</li>
    	    </ol></li>
    	    
    	    <li>Verify forward to MouseBlast of protein sequences
    		<p concordion:assertTrue="true">PASS - jlewis 8/15/2013</p> 
    		<ol>
    		   <li>Forward a single row for a family with more than one member and one that has sequences for all 3 organisms, PIRSF001199.</li>
    		   <li>Forward 2 rows of a family page with more than 2 rows, PIRSF001873.</li>
    		   <li>Forward all rows for a family with 3 or more members, PIRSF002051.</li>
    		   <li>Forward all rows for a family with just 1 member, PIRSF002089.</li>
    		   <li>Forward all rows for a family with a member that has no select box, PIRSF005491.</li>
    		   
    		   <li>Forward an Ensembl protein sequence, PIRSF006060</li>
    	    </ol></li>
    	    
    	    <li>Verify page display, links, pop-ups, etc for all supported browsers (IE, firefox, safari, chrome)
    		<p concordion:assertTrue="true">PASS - jlewis 8/15/2013</p> 
    		<ol>
    		   <li>Display PIRSF001015.  Verify display, links, pop-ups, etc.</li>
    	    </ol></li>
		</ol>
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>