<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>US 177 Marker MP Sum Download</title>
</head>

<body>

<div id="maintitle"><h1>Marker MP Summaries Download</h1>
    <ul>
        <li>US 177 - Add download option - Phenotype Summaries</li>
    </ul>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
    <ol> 
       <li>Add Excel download option to the 2 new MP Summaries for a marker (rolled up ones and multigenic)</li>
       <li>Download button -- use label and styling in place elsewhere in the WI</li>
       <li>Normalize the data found on this report and include 1 row for each set of unique values - all single values</li>
       <li>Display the following columns in the report in the following order (label:definition)
       <ol><li>Genotype: allele pair(s)</li>
           <li>Genetic Background: strain background</li>
           <li>Genotype ID: MGI id</li>
           <li>Qualifier: e.g. normal</li>
           <li>Phenotype: MP term name</li>
           <li>Phenotype Summary Category: slim grid header</li>
           <li>Reference: J number</li>
       </ol></li>
       <li>The following fields may have multiple values -- which will result in multiple rows in the file
       <ol><li>Phenotype Summary Category (slim grid header)</li>
           <li>Reference (J number)</li>
       </ol></li>
       <li>The Phenotype Summary Category (slim grid header) may also be blank</li>
    </ol>   
   
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
		    <li>Verify genotypes in download file 
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Get Excel file for summary with only single allele pairs; cw.  Expect: Same allele pais as on Summary page. </li>
    		   <li>Get Excel file for summary with multiple allele pairs; Rb1 (multigenic).  Expect: Same allele pairs as on Summary page.</li>
    		   <li>Get Excel file for summary with transgene alleles; Trp53.  Expect: Same as summary</li>
    		   
    		</ol></li>
    		
    		<li>Verify genetic background in download file
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Get Excel file where backgrounds have many special characters; Pax4  Expect: all special characters displayed.</li>
    		   <li>Get Excel file for summary with background superscripts; Kras.  Expect: superscripts in pointy brackets -- embedded and trailing superscripts for this summary.</li>
    		   
    		</ol></li>
    		
    		<li>Verify normal qualifiers
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Get Excel file for summary with normal qualifiers; Ncoa2.  Expect: 4 rows with normal. </li>
    		   
    		</ol></li>
    		
    		<li>Verify that multiple J numbers result in multiple rows
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Get Excel file for a summary with multiple J numbers; Apoe, term= atherosclerotic lesions.  Expect: Term listed 3 times for 1 genotype - why? (6 refs, 2 refs, 1 ref)</li>
    		   <li>Get Excel file for a summary with multiple J numbers and slim terms for same annotation, Hfm, term=small ears.  Expect: term has 3 header terms and 2 references = 6 rows for term. </li>
    		   <li>Get Excel file with multiple J numbers for a normal term; Apoe, term = nervous system.  Expect: nervous system in 2 rows, each with normal qualifier.</li>
    		   
    		</ol></li>
		  
		    <li>Verify slim term header column
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Get Excel file for a summary with complete coverage of all slims; Trp53 and Bmpr1a.  Expect:  all headers represented between the 2 genes.</li>
    		   <li>Get Excel file for a summary with terms not in the slim grid; Myo5a (normal phenotype) or Atox1 (other phenotype).  Expect: field to be blank.</li>
    		   <li>Get Excel file for a summary with term with multiple headers; Sec23b, term=nasal gland degneration.  Expect: headers=craniofacial, endocrine/exocrine, growth/size/body, respiratory; term in 3 genotypes; 12 rows for this term.</li>
    		   
    		</ol></li>
    		
    		<li>Verify downloads for atypical summaries
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Get Excel file with only 1 row; Atp10a or Zbp1.  Expect: 1 row in file.</li>
    		   <li>Get Excel file with only 1 genotype, but multiple terms; Olfr6.  Expect: 2 rows (1 row per term).</li>
    		   <li>Get Excel file with only 1 term, but with multi-value fields; Xxx.  Expect: multiple rows in file.</li>
    		   <li>Get Excel file with no slim terms (all blank); Prrx2.  Expect: 3 rows with no slim term headers.</li>
    		   
    		</ol></li>
		
		     <li>Verify download option works on all supported browers: Firefox, Chrome (PC, Mac); Safaric (Mac)
    		<p concordion:assertTrue="false">new test</p>   
    		<ol>
    		  <li>Download a file on each of the supported browsers.  </li>
    		  
    		</ol></li>
    		
		</ol>
    	    
      	    
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>