<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 119</title>
</head>

<body>

<div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>

<div id="maintitle"><h1>Load B38 gene models & associated data in Prod WI</h1>
    <ul>
        <li>User Story 119, Load B38 gene models, coordinates, transcripts, proteins, raw biotypes, new markers, marker associations in Prod WI.</li>
    </ul>
</div>

<div id="acceptance">
    <h2>Test Requirements</h2>
	<ul>
		<li>Verify all Build 37 gene model data is gone.
		    <ul><li>Gene models, coordinates, biotypes, transcripts, proteins, marker associations</li>
		        <li>deletion of directly assigned coordinates is NOT part of this story</li>
		        <li>deletion of SNP and GT associations is NOT part of this story</li>
		        <li>other Build 37 data/associations is NOT part of this story</li>
		    </ul>
		<li>Verify new markers are created for NCBI, VEGA, and Ensembl.</li>    
	    <li>Verify Gene Models & their coordinates are loaded for NCBI, VEGA, and Ensembl for existing and new markers.</li>
	    <li>Verify transcripts and proteins are loaded for the Gene Models</li>
	    <li>Verify raw biotypes are loaded and translated properly for the Gene Models</li>
	    <li>Verify the Representative Sequence Algorithm selects the appropriate gene model</li>
	    <li>Verify evidence links display and link to the appropriate page</li>
	    <li>Verify that cM locations from B37 are kept in place (recalculation and/or new calculation turned off)</li>
	    <li>Verify Build 38 is displayed in front-end products (all WIs, MouseBlast)</li>
	    <li>Exclude - links to external genome browsers & gbrowse are not part of this story</li>
	</ul>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		<ol>
    		<li>Verify new markers and their gene models are created for B38 in the database  
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Query for marker <b>Gm21711</b>, this is a new novel gene from Ensembl
    			<ul>Expected results (line from Sophia's file): <b>Gm21711	predicted gene, 21711	8	In Progress	J:91388			ENSMUSG00000096123 is on chr8:128320947-128322648, - strand. ENSEMBL annotation of mouse genome build 38, Gene type: lincRNA
    			</b></ul></li>
    			
    			<li>No novel genes from VEGA for Build 38 (confirmed with Sophia)</li>
    			
    			<li>Query for marker <b>Gm20800</b>, this is a new novel pseudogene from NCBI
    			<ul>Expected results (line from Sophia's file): <b>Pseudogene	Gm20800	predicted gene, 20800	Y	In Progress	J:90438			100039357 LOC100039357 Y-linked testis-specific protein 1-like on Chr Y</b>
    			</ul></li>
    			    			
       		</ol></li>
       		
       		<li>Verify updated gene models for existing markers for Build 38 are updated in the database
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    		    <li>Query for marker <b>Brca2</b>, this is an existing gene with an updated gene model from Ensembl
    			<ul>Expected results (line from Sophia's file): <b>ENSMUSG00000041147	5	150522630	150569746	+	chr5:150522630-150569746, + strand. ENSEMBL annotation of mouse genome build 38, Gene type: protein_coding; Gene Name: Brca2</b>
    			</ul></li>    		
    		
                <li>Query for marker <b>Vmn2r-ps1</b>, this is an existing pseudogene with a new gene model from Ensembl for B38
    			<ul>Expected results: <b>ENSMUSG00000090261</b>
    			</ul></li>    	   		
    		
    		    <li>Query for marker <b>Trp53</b>, this is an existing gene with an updated gene model from VEGA
    			<ul>Expected results (line from Sophia's file): <b>OTTMUSG00000005982	11	69580359	69591873	+	chr11:69580359-69591873, + strand. VEGA annotation of mouse genome build 38, Gene type: _protein_coding; Gene Name: Trp53 </b>
    			</ul></li>
    			
    			<li>Query for marker <b>Sry</b>, this is an existing gene with an updated gene model from NCBI
    			<ul>Expected results (line from Sophia's file): <b>21674	Y	2662471	2663658	-	chrY:2662471-2663658, - strand. NCBI annotation of mouse genome build 38, GENE: Sry</b>
    			</ul></li>
    				
       		</ol></li>
       		
       		<li>Verify gene models deleted for Build 38 are no longer in the database
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    		    <li>Query for gene model <b>ENSMUSG00000072491</b>, this is an existing B37 gene model from Ensembl
    			<ul>Expected results: <b>not found</b>
    			</ul></li>
    			
    			<li>Query for gene model <b>OTTMUSG00000037917</b>, this is an existing B37 gene model from VEGA
    			<ul>Expected results: <b>not found</b>
    			</ul></li>
    			
    			<li>Query for gene model <b>236598</b>, this is an existing B37 gene model from NCBI
    			<ul>Expected results: <b>not found</b>
    			</ul></li>
    				
       		</ol></li>
       		
       		<li>Verify gene models that have associations to more than 1 marker. 
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    		    <li>Use the QuickSearch and query for <b>ENSMUSG00000007440</b>; view markers returned for this <b>Ensembl</b> gene model
    			<ul>Expected results: <b>Pcdha1, Pcdha2, Pcdha4, Pcdha5, Pcdha10, Pcdha11, Pcdha12, Pcdhac2</b>
    			</ul></li>
    			
    			<li>No test case for <b>VEGA</b> that I could find looking at files provided by Sophia.
    			</li>
    			
    			<li>Use the QuickSearch and query for gene model <b>100503456</b>, view markers returned for this <b>NCBI</b> gene model.
    			<ul>Expected results: <b>Gm11342, Gm11345</b>
    			</ul></li>
    			
       		</ol></li>
       		
       		<li>Verify markers that are associated to more than 1 gene model.
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    		    <li>Display the Marker Detail for marker <b>Qk</b>; view associations to <b>Ensembl</b> gene models. 
    			<ul>Expected results: <b>ENS00000044407, ENS00000062078</b>
    			</ul></li>
    			
    			<li>Display the Marker Detail for marker <b>Gm6541</b>; view associations to <b>VEGA</b> gene models.
    			<ul>Expected results: <b>OTT00000036694, OTT00000036405</b>
    			</ul></li>
    			
    			<li>Display the Marker Detail for marker <b>Gab3</b>; view associations to <b>NCBI</b> gene models
    			<ul>Expected results: <b>100503280</b> (for Build 37 an add'l model was listed here - 210710)
    			</ul></li>
    			
       		</ol></li>
       		
    		<li>Verify "Evidence" links are displayed on Marker Detail page and the link goes to Build 38 data 
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Query for marker <b>Pax6</b>, this marker has evidence for all 3 providers: Ensembl, VEGA, and NCBI
    			<ul>Expected results: <b>Links in the "Other database links" section of the Marker Detail page for each provider.  Link works and goes to B38 evidence.</b>
    			</ul></li>
    			
       		</ol></li>
       		
       		<li>Verify biotype mismatch identified correctly and displayed 
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Query for marker <b>Tead1</b>; Display Biotype conflict text
    			<ul>Expected results: <b>conflict with Ensembl and VEGA</b>
    			</ul></li>
    						
       		</ol></li>
       		
    		<li>Verify Rep Sequence Algorithm works as expected
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Query for marker <b>Brca</b>
    			<ul>Expected results:  Rep genomic gene model is <b>Ensembl</b>
    			</ul></li>
    			
    			<li>Query for marker <b>Trp53</b> 
    			<ul>Expected results: Rep genomic gene model is <b>VEGA</b>
    			</ul></li>
    			
    			<li>Query for marker <b>Sry</b>
    			<ul>Expected results: Rep genomic gene model is <b>NCBI</b>
    			</ul></li>			  			
       		</ol></li>
    		
    		<li>Verify associated Build 38 Proteins/Transcripts are loaded and associated to the appropriate marker for Ensembl & VEGA
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Query for marker <b>Kit</b>; Display Sequence Summary for Kit;
    			<ul>Expected results: <b>Proteins and Transcripts for VEGA and Ensembl included in the list</b>
    			</ul></li>
    			
       		</ol></li>
    		
    		<li>Verify cM positions unchanged from B37 calculations
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    		    <li>Query for marker <b>Lepr</b> (a marker that existed in B37)
    			<ul>Expected results:  <b>46.96 cM</b>
    			</ul></li>
    			
    			<li>Query for marker <b>Gm21711</b> (a new marker in B38) 
    			<ul>Expected results: <b>Syntenic</b>
    			</ul></li>
    			
    			<li>Query for marker <b>Avp-rs1</b> (a marker with UN chromosome)
    			<ul>Expected results: <b>nothing displayed for cM</b>
    			</ul></li>			
    		
    			<li>Query for marker <b>DXYCbl3</b>, <b>mt-Nd5</b>, <b>DYBish6</b> (markers on XY, MT, Y)
    			<ul>Expected results: <b>Syntenic</b>
    			</ul></li> 			
       		</ol></li>
    		
    		<li>Verify build text has changed to Build 38 in PROD WI and MouseBlast
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Python WI - Marker Summary (web, tab, excel); SNP Summary</li>   
    			<li>MouseBlast - Query Form</li>
    			<li>Java WI - Marker, Allele, GXD, GXD Expanded, SNP Query Forms; Marker Detail; Allele Detail; SNP Detail; Batch Query</li> 			
       		</ol></li>
    		
    		
		</ol>
	</div>
	<div id="sql">
	
	<h3>SQL tests</h3>
		<ol>
    	<li>Quality/Sanity check of markers associated with gene models from the different gene model providers.
    	<ol><li type=A>Counts of markers and associated gene models/transcripts/proteins.
    	
    	
    		<blockquote><font size=-1>Comparing counts of markers and associated gene models/transcripts/proteins between DEV_MGI..mgd and the C4AM Test area (DEV1_MGI..mgd_wi1) tests the overall success of the gene model loads and gene model association loads.  The distribution of counts among MGI Feature Types provides a quick check that in general, the appropriate marker types were associated with gene models, and with corresponding transcripts and proteins.  All Feature Types are included, even if there are no markers of that Feature Type associated with gene models.  This assures that all markers with a Feature Type are accounted for.</font></blockquote>
 
    		<ol>
    			<li type=1>VEGA_GM_Report<br>
    			Backend sybase database (DEV1_MGI..mgd_wi1)</li>
    			<p concordion:assertTrue="false">PASS 9/9/12 rmb</p>
    			<blockquote><font size= -1>For each Feature type, the following information is provided:<br/>
- Total number of markers with or without a gene model association (Markers_Total)<br/>
- Total number of markers associated with at least one VEGA Gene Model (Markers_VEGA_GM)<br/>
- Total number of VEGA Gene Models (VEGA_GeneModels)<br/>
- Total number of markers associated with at least one VEGA Transcirpt (Markers_VEGA_T)<br/>
- Total number of VEGA Transcripts (VEGA_Transcripts)<br/>
- Total number of markers associated with at least one VEGA Protein (Markers_VEGA_P)<br/>
- Total number of VEGA Transcripts (VEGA_Proteins)</font></blockquote>
    			<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
    		 	&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;B37 DEV_MGI..mgd Results: <a href="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US119_VEGA_GM_Report.xlsx">US119_VEGA_GM_Report.xlsx</a>
    			<br>
    			&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<font size= -1>Note: The SQL for this report is included on a separate tab in file US119_VEGA_GM_Report.xlsx</font>
    			</ul><br>
    			
    			<li type=1>Ensembl _GM_Report<br>
    			Backend sybase database (DEV1_MGI..mgd_wi1)</li>
    			<p concordion:assertTrue="false">PASS 9/9/12 rmb</p>
    			<blockquote><font size= -1>For each Feature type, the following information is provided:<br>
- Total number of markers with or without a gene model association (Markers_Total)<br>
- Total number of markers associated with at least one Ensembl Gene Model (Markers_Ensembl_GM)<br>
- Total number of Ensembl Gene Models (Ensembl_GeneModels)<br>
- Total number of markers associated with at least one Ensembl Transcirpt (Markers_Ensembl_T)<br>
- Total number of Ensembl Transcripts (Ensembl_Transcripts)<br>
- Total number of markers associated with at least one Ensembl Protein (Markers_Ensembl_P)<br>
- Total number of Ensembl Transcripts (Ensembl_Proteins)</font></blockquote>
    			<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
    		 	&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;B37 DEV_MGI..mgd Results: <a href="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US119_VEGA_GM_Report.xlsx">US119_Ensembl_GM_Report.xlsx</a>
    		 	<br>
    		 	&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<font size= -1>Note: The SQL for this report is included on a separate tab in file US119_Ensembl_GM_Report.xlsx</font>
    		 	</ul><br>

<li type=1>NCBI _GM_Report<br>
Backend sybase database (DEV1_MGI..mgd_wi1)</li>
  <p concordion:assertTrue="false">FAIL 9/9/12 rmb<br>
  No NCBI Gene Models were loaded.</p></li>
<blockquote><font size= -1>For each Feature type, the following information is provided:<br>
- Total number of markers with or without a gene model association (Markers_Total)<br>
- Total number of markers associated with at least one NCBI Gene Model (Markers_NCBI_GM)<br>
- Total number of NCBI Gene Models (NCBI_GeneModels)<br>
- Total number of markers associated with at least one NCBI RefSeq Transcript (Markers_RefSeq_T)<br>
- Total number of NCBI RefSeq Transcripts (RefSeq_Transcripts)<br>
- Total number of markers associated with at least one NCBI RefSeq Protein (Markers_RefSeq_P)<br>
- Total number of NCBI Transcripts (Ensembl_Proteins)
</font></blockquote>
    <ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;B37 DEV_MGI..mgd Results: <a href="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US119_VEGA_GM_Report.xlsx">US119_NCBI_GM_Report.xlsx</a>
<br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<font size= -1>Note: The SQL for this report is included on a separate tab in file US119_NCBI_GM_Report.xlsx</font>
</li>

<li type=A>Gene Model Cardinality counts
<ol>
    			<li type="1">Markers associated with multiple GM IDs<br/>
    			Backend sybase database (DEV1_MGI..mgd_wi1)</li>
    			<p concordion:assertTrue="false">PASS 9/9/12 rmb</p>
    			<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li></ul>
    		
       		</ol>
       		</ol>

 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>