<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 23</title>
</head>

<body>

<div id="maintitle"><h1>Disease Detail - use new roll-up annotations</h1>
    <ul>
        <li>User Story 23 - Change disease detail to use pre-computed gene rollup annotations</li>
    </ul>
</div><div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>



<div id="acceptance">
    <h2>Test Requirements - Disease Detail & All Mouse Models page</h2>
	<ol>
	<h3>Disease Detail</h3>
	   <li>The basic display of the Disease Detail does not change.  The new roll-up rules alter the categorization of the markers and their models.  Markers may move to other sections on the page, be removed from the page, or be added to the page.</li>
	   <li>The following ribbons on the page should remain the same (data content and display)
	   <ol><li>Human Disease</li>
	       <li>Synonyms</li>
	       <li>View all models</li>
	       <li>References</li></ol>
	   <li>The basic rules for display, text, links, homologs, etc. of the Genes and mouse models ribbon does not change.
	   <ul><li>There are new rules to determine which category to display (or not) a mouse gene.   The rules have been re-implemented so need to test all cases.
	       <ol><li>Roll-up to a gene for OMIM annotations to a simple 1 marker (where marker type = gene) genotype.</li>
	           <li>Roll-up to a gene for OMIM annotations to a conditional genotype that has just 1 marker (where marker type = gene) when recombinase, transgenic reporter, and wild type alleles are factored out.</li>
	           <li>Roll-up to a gene for OMIM annotations to a complex genotype that has just 1 marker (where marker type = gene) when transgenic reporters are factored out.</li>
	           <li>Remove Gt(ROSA)26Sor from the rolled up set of genes.</li>
	           <li>Remove genes that have only NOT annotations from the rolled up set.</li>
	           <li>Note: differences with HMDC - slash alleles not removed; allele/disease annotations not included; transgenic reporter bug</li>
	       </ol></li>
	       <li>Rules for the display of Human genes have not changed -- human phenotypic markers should not be displayed.</li>
	   </ul></li>
	   <li>The basic rules for display, text, links of the Transgenes and other genome features ribbon does not change.
	   <ul><li>There are new rules to determine which category to display (or not) a transgene or other non-gene marker.
	       <ol><li>Same rules as above except for non-gene markers.</li></ol></li>
	   </ul></li>
	   <li>Mouse Models column pop-up - the rules to determine what models are included for each marker does not change.</li>
	   
	<h3>All Mouse Models page</h3>
	 <li>The basic display of the All Mouse Models page does not change.  The total number of models does not change.  The new roll-up rules alter the categorization of the models.  </li>
	 <li>The Disease phenotype not observed ribbon on the page should remain the same (data content and display).</li>
     <li>The basic rules for display, text, links, etc. of the Mouse models ribbon does not change.
	   <ul><li>There are new rules to determine which category to display each Mouse Model.
	       <ol><li>Same rules as above - both gene rules and other marker type rules.</li></ol></li>
	   </ul></li>	   
	   
	   
	</ol>
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
    		<li>Disease Detail page - diseases with no mouse models
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		  <li>Display disease with no models & no human marker associations; 603047 - Astigmatism.  Expect: 0 mouse or human markers.</li>
    		  <li>Display disease with no models, but with human marker associations; 603956 - Cervical Cancer.  Expect: 1 human gene (FGFR3)</li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases where the results are unchanged with new roll-up rules
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		  <li>Display disease with model(s) for 1 gene; 600321 - Diabetes...IDDM7.  Expect: 1 mouse gene (Neurod1).  HMDC differs - add'l human gene (human phenotypic marker).</li>
    		   <li>Display disease with model(s) for multiple mouse genes; 109100 - Autoimmune Disease.  Expect: 2 mouse genes (Cd200, Gpr132). HMDC differs - 3 add'l genes due to Allele/Disease annotations.</li>
    		   <li>Display disease with models for many mouse genes & many associated human markers; 600807 - Asthma.  Expect: 4 mouse genes; 13 human genes</li>
    		   <li>Display disease with model(s) for 1 non-gene; 188550 - Thyroid Carcinoma.  Expect: 1 other; 7 human genes.  HMDC differs - add'l 2 others due to Allele/Disease annotation. </li>
    		   <li>Display disease with model(s) for many non-genes; 607136 - Spinocerebellar Ataxia 17; SCA17.  Expect: 4 mouse others; 1 human gene.</li>
    		   <li>Display disease with model(s) for many genes and many non-genes; 152700 - Lupus; SLE.  Expect: 26 mouse genes; 5 mouse others; 6 human genes.</li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases with genes that have n-m homologs
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		   <li>Display disease with 1-m homology class; 253550 - Spinal Muscular Atrophy, Type II; SMA2. Expect: 1 mouse gene; 1 human gene.</li>
    		   <li>Display disease with n-1 homology class; 603903 - Sickle Cell Anemia. Expect: 1 mouse genes; 0 others; 1 human gene. HMDC differs - add'l 4 mouse others due to allele/disease annotations.</li>
    		   <li>Display disease with 0-1 and 1-0 homology classes; 181500 - Schizophrenia; SCZD.  Expect: 28 mouse genes; 6 others; 15 human genes.</li>
    		   <li>Display disease with human markers that have no homology class; 130650 - Beckwith-Wiedemann Syndrome; BWS.  Expect: 2 mouse genes; 1 others; 6 human genes.  2 human genes are not in a homology class.</li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases with conditional genotype models
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		   <li>Display disease that lost gene due to 1 marker rule for conditionals; 114500 - Colorectal Cancer.  Expect: 1 mouse; 0 other; 4 human.  Lost Kras.</li>
    		   <li>Display disease that includes a gene due to conditional rule; 142340 - Diaphragmatic Hernia, Congenital. Expect: 10 mouse; 0 other; 0 human.  Nr2f2 included due to recombinase allele factored out. Nkx3-2 is the marker for the recomb allele (and should not be included). HMDC differs: Includes a phenotypic human marker; DIH1.</li>
    		   <li>Display disease that changed from 2 genes with models to 0 genes w/ models due to 1 marker rules for conditionals; 606889 - Alzheimer Disease 4.  Expect: 0 mouse genes; 1 other; 2 human. Lost Psen1, Psen2. Psen2 models will display in human only section.</li>
    		   <li>Display disease w/ associated model for a simple genotype for a recombinase allele; 125800, Diabetes Insip...Autosomal.  Expect: model to be included with other models for Aqp2 on main page and All Models page. (Public doesn't handle this model type properly, but new rules fix it.)</li>
    		   <li>Display disease w/ associated model for a recombinase transgene allele; 114480 - Breast Cancer.  Expect: this cre Transgene to be listed in Other section. (new other marker -- Tg(MMTV-Erbb2*,-cre)1Mul). (Public doesn't handle this marker/model properly, but new rules fix it.)
    		   <ul><li>Mouse = 9 (lost 4: Bin1, Ctnnb1, Pten, Trp53); Human = 24; Other = 22 (gained 1 (above); lost 2: Tg(CAG-Has2)99Nita & Tg(MMTV-Myc)141-3Led)</ul></li>
    		   <li>Display disease that includes a non-gene marker due to conditional rule; 612242 - Chromosome 10q23 Deletion Syndrome.  Expect: 0 mouse; 1 other; 0 human.  Marker included when a recombinase allele is factored out. (NOTE: disease reference ribbon now present - fix from Public)
    		   <li>Display disease that lost mouse non-gene models; 176807 - Prostate Cancer.  Expect: 8 mouse; 11 other; 12 human.  
    		   <ul><li>Lost 8 mouse genes: Ar, Brca2, Braf, Scrib, Spry1, Spry2, Tert, Trp53.  Lost 1 other: Tg(ARR2/Pbsn-FGF8)3Prb))</li></ul></li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases with complex genotype models
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		   <li>Display disease w/ associated complex genotype model and other models; 211980 - Lung Cancer.  Expect: Col1a1 not listed on main page, but model included on All Models page in Add'l Complex section.</li>
    		   <li>Display disease w/ only complex genotype models; 605814 - Citrullinemia, Type II, Neonatal-Onset.  Expect: No mouse genes listed on main page, but models included on All Models page.</li>
    		   <li>Display disease w/ only complex genotype models and also no human genes; 160750 - Myositis. Expect no mouse/human gene list on main page, but models included on All Models page.</li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases with transgenic reporter models
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		   <li>Display disease w/ associated complex model w/ transgenic reporter allele that factors to simple geno;166950 - Teratoma, Ovarian.  Expect: 1 mouse; 1 other; 0 human. Gained Foxo3.</li>
    		   <li>Display disease w/ associated model for a transgenic reporter allele with a simple genotype; 615538 - Chromosome 22q13 Duplication Syndrome.  Expect Tg(Shank3-EGFP)1Hzo listed in the Other section with 1 model.</li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases with Gt(ROSA) genotype models
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		   <li>Display disease that has a Gt(ROSA) model; 300813 - Sarcoma, Synovial.  Expect: Gt(ROSA)26Sor to not be included in the gene list.  Plus - appropriate disease references now displayed.</li>
    		   <li>Display disease that has a conditional model where the recombinase Gt(ROSA) allele; 614286 - Myelodysplastic; MDS.  Expect: Bap1 to be in the gene list, but not Gt(ROSA) (the recombinase allele).</li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases with slash allele genotype models
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		  <li>Display disease with model(s) associated with slash alleles; 183600 - Split-Hand/Foot Malformation 1: SHFM1.  Expect: 2 mouse genes; 0 human genes & 2 disease references vs. none.  HMDC differs - 0 mouse genes due to slash alleles being removed. </li>
    	    </ol></li>
    	    
    	    <li>Disease Detail page - diseases with NOT models
    		<p concordion:assertTrue="true">PASS - Jon's dev version</p>  
    		<ol>
    		   <li>Display disease with only NOT models; 208230 - Arthropathy, Progressive Pseudorheumatoid, of Childhood; PPAC.  Expect: 0 mouse; 0 other; 1 human. NOT models displayed as before. </li>
    		   <li>Display disease where a gene has only NOT models and should not be on the roll-up list; 143100 - Huntington Disease; HD.  Expect: 4 mouse; 21 other; 1 human.  Gene Hap1 has only NOT models & is not on gene list. HMDC differs: Tg(Camk2a-tTA)1Mmay included due to transgenic reporter bug.</li>
    		   <li>Display disease NOT models in Human section; 208230 - Arthropathy, Progressive Pseudorheumatoid, of Childhood; PPAC.  Expect: 0 mouse; 0 other; 1 human.  2 NOT models.</li>
    		   <li>Display disease with NOT models in Gene section; 607822 - Alzheimer Disease 3.  Expect: 1 mouse; 1 other; 1 human.  Mouse Psen1 has 1 NOT model.</li>
    		   <li>Display disease with NOT models in non-Gene section; 614266 - Barrett Esophagus.  Expect: 0 mouse; 1 other; 3 human.  Other marker has model and NOT model.</li>
    	    </ol></li>
    		
    	    
    	    
  
		</ol>
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>