<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 12</title>
</head>

<body>

<div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>

<div id="maintitle"><h1>Update QuickSearch to new homology</h1>
    <ul>
        <li>User Story 12, Update Quick Search to use new n-m homology classes</li>
    </ul>
</div>

<div id="acceptance">
    <h2>Test Requirements</h2>
	<ol>
	   <li>The QuickSearch returns mouse markers by matching non-mouse nomenclature.  The results that these matches return are based on associations from newly loaded n-m homology classes from HomoloGene.
	   <ol><li>Non-mouse nomenclature includes: non-mouse symbols (loaded species); non-mouse names (loaded species); human and rat synonyms.  Loaded species: human, rat, cattle, chimpanzee, cow, dog.</li>
	       <li>A match to a non-mouse marker via nomenclature effectively matches a Homology Class.  So, return all mouse markers in that class in the Marker bucket
	       <ul><li>Example: "RG" is a synonym of human C4A. If the user enters this, they would get both mouse C4a and C4b returned because these genes are in the same Homology Class.  The reason would be "Human Synonym: RG".</li>
	       </ul></li>
	       <li>A nomenclature match to a non-mouse marker may result in no matches to mouse markers
	       <ol><li>A non-mouse Marker may match a Homology Class that has no mouse markers.</li>
	           <li>If the matching nomen/synonyms only match non-mouse genes not in a Homology Class, then no mouse markers are matched or returned because of homology</li>
	       </ol></li>
	       <li>Nomenclature matches by<u> mouse nomenclature</u> will NOT result in any add'l results due to Homology Class relationships.
	       <ul><li>Example:  "Slp" is a synonym of mouse C4a, if the user enters "Slp", we'd return C4a because of mouse synonym, but NOT C4b</li>
           </ul></li>
	   </ol></li>
       
       <li>Update display for non-mouse marker "why did this match" labels.
       <ol><li>The "why matched" text continues to display the type of nomenclature match -- but remove the word <i>ortholog</i> for all the labels.</li>
	       <li>Change display of <i>dog, domestic</i> to <i>dog</i>.</li>        
       </ol></li>
       
       <li>Return results for a match to a HomoloGene ID.
       <ol><li>Return all mouse markers in that class, if any, in the marker bucket with a reason, eg. <i>ID: 36030 (HomoloGene)</i></li>
           <li>Return a single result in the ID bucket with a link to the new Homology Class page
           <ol><li>Display the 3 columns: <i>Homology; Class with </i>(species list w/ counts)<i>; ID: 36030 (HomoloGene)</i></li></ol></li>
       </ol></li>
       
       <li>Return results for a match to a non-mouse ID (e.g EntrezGene, HGNC, OMIM gene id, RefSeq, UniProt, etc.)
       <ol><li>A match to a non-mouse ID implies a match to a HomoloGene Class.  Return all mouse markers, if any, included in that class to the marker bucket and display 
               the reason for the match as it is now, eg. <i>ID: 720 (EntrezGene - Human)</i></li>
           <li>In the ID bucket return a link to the Homology Class page.
           <ol><li>Display the 3 columns: <i>Homology</i>; (non-mouse symbol, name); <i>ID: 720 (EntrezGene - Human)</i>.</li></ol></li>
	  </ol>
 </ol>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		<ol>
    		
    		<li>Test 1
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>test 1.1</li>
    			<li>test 1.2 </li>	
       		</ol></li>
    		
    		
		</ol>
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>