<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>US 85: HMDC - femover refactoring</title>
</head>

<body>

<div id="maintitle"><h1>HMDC - Refactor femover</h1>
    <ul>
        <li>US 85: HMDC - femover refactoring</li>
    </ul>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
    <ol> 
       <li>This story breaks a large HMDC gatherer in the femover into smaller ones.
       <ol><li>Genocluster HMDC annotations: hdp_genocluster_genotype, hdp_genocluster, hdp_genocluster_annotation</li>
           <li>Disease references: hdp_term_to_reference</li>
           <li>Homology relationships: hdp_gridcluster, hdp_gridcluster_marker</li>
           <li>Marker/genotype HMDC annotations: hdp_annotation</li>
           <li>Marker references: hdp_marker_to_reference (done previously - no change)</li>
       </ol>
       <li>SEARCH by GENE nomenclature (symbols, names, IDs) or location (mouse or human coordinates)
       <ul><li>GRID TAB
           <ol><li>return MP/OMIM annotations of ALL simple mouse genoclusters for matching markers.</li>
               <li>return all human OMIM annotations for matching genes
               <li>return orthologs (and associated annotations) of matching genes (i.e. match to mouse marker; return mouse plus human ortholog OR match to human gene; return human plus mouse ortholog)</li>
            </ol></li>
            <li>GENE TAB 
            <ol><li>return all matching mouse and human markers plus their orthologs -- regardless of the presence of MP and/or OMIM annotations.
                  <ul><li>human genes returned do NOT have to have a homology relationship</li></ul></li>
                <li>for each mouse or human marker returned display OMIM diseases rolled up to</li>
                <li>for each mouse marker returned display high-level MP terms rolled up to</li>
                <li>for each mouse marker returned display reference counts: all and disease relevant.</li>
            </ol></li>
            <li>DISEASE TAB
           <ol><li>return all diseases that are present on the grid.</li>
               <li>for each disease display mouse model count - all models including those for complex genotypes</li>
               <li>for each disease display associated mouse genes - only those that roll up.</li>
               <li>for each disease display associated human genes</li>
               <li>for each disease display mouse model references</li>
           </ol></li>
          
       </ul></li>
       <li>SEARCH by Phenotype and Disease terms or IDs (OMIM, MP)
       <ul><li>GRID TAB
           <ol><li>return all MP/OMIM mouse annotations for simple genoclusters that contain a match to the search string</li>
               <li>return all OMIM annotations for human genes that match via an OMIM term.</li>
               <li>group results by mouse/human orthologs -- but don't include an ortholog's annotations unless specifically matched.</li>
            </ol></li>
            <li>GENE TAB 
            <ol><li>return mouse genes that have an annotation that matches the search string.  The annotation must be to a simple genocluster (complex annotations are not searched).</li>
                <li>return human genes that have an OMIM annotation that matches the search string.</li>
                <li>Orthologs for the above genes are NOT included (different than search by genes above).</li>
                <li>for each mouse or human marker returned display rolled up OMIM diseases</li>
                <li>for each mouse marker returned display rolled up high-level MP terms</li>
                <li>for each mouse marker returned display reference counts: all and disease relevant.</li>
            </ol></li>
            <li>DISEASE TAB
           <ol><li>return all diseases that are present on the grid.</li>
               <li>return diseases that match the search string, but that may not be on the grid.
                   <ul><li>Search by OMIM id with no annotations OR search by text.  E.g. "Lung Cancer".</li></ul></li>
               <li>for each disease display mouse model count - all models including those for complex genotypes</li>
               <li>for each disease display associated mouse genes - only those that roll up.</li>
               <li>for each disease display associated human genes</li>
               <li>for each disease display mouse model references</li>
           </ol></li>
          
       </ul></li>
       
       
       
    </ol>   
    
   
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
		
		    <li>Verify search by Gene Nomenclature returns the correct data on the Grid Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by text that matches a mouse gene symbol; Pax4.  Expect: all simple genocluster annotations for gene plus all human ortholog OMIM annotations.</li>
    		    <li>Search by text that matches a human gene symbol; TP53.  Expect: all human OMIM annotations for human gene plus all simple genocluster annotations for mouse ortholog (Trp53).</li>
    		    <li>Search by text that matches a mouse gene ID; MGI:96677 (Kit).  Expect: 1 row for Kit - all simple genotypes plus Human ortholog (KIT) OMIM annotations.</li>
    		    <li>Search by text that matches a human gene ID; HGNC:6342 (KIT).  Expect: same results as mouse Kit.</li>
    		    </ol></li>
    		
    		<li>Verify search by Gene Nomenclature returns the correct data on the Gene Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by text that matches a mouse/human homology pair; Brca1.  Expect: Both Brca1 and BRCA1 on gene tab.</li>
    		    <li>Search by text that matches only mouse symbol of homology pair; Tomt.  Expect: Tomt and LRTOMT on gene tab (note this is a human ortholog of non-hybrid relationship).</li>
    		    <li>Search by text that matches only human symbol of homology pair; RGPD1.  Expect: Ranbp2 and the 7 human orthologs per H-gene. (RGPD1...RGPD6, RGPD8)</li>
    		    <li>Search by text that matches mouse symbols without MP/OMIM; Ren2.  Expect: Mouse symbols Iap1-1 displayed on Gene tab and not on Grid tab.</li>
    		    <li>Search by text that matches human symbols without OMIM; Pax6.  Expect:  Human symbols PAUPAR displayed on Gene tab and not on Grid tab.</li>
    		   </ol></li>
    		   
    		<li>Verify search by Gene Nomenclature returns the correct data on the Disease Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by text that matches mouse and human genes with OMIM; pten.  Expect: 19 diseases on grid to be on Disease tab.</li>
    		    <li>Search by text that matches only mouse genes with OMIM; Sox7.  Expect: 1 disease on grid to be on Disease tab.</li>
    		    <li>Search by text that matches only human genes with OMIM; PAX4.  Expect: 3 diseases on grid to be on Disease tab.</li>
    		    <li>Search by text that matches a disease cluster; C4a.  Expect: 2 disease columns on Grid; 3 diseases on Disease tab.</li>
    		   </ol></li>
    		   
    		<li>Verify search by Gene Nomenclature recognizes correct homology relationship for Grid and Gene Tabs.
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by text that matches a gene with only a HomoloGene relationship: Selm.  Expect: 1 row for Selm, SELM.</li>
    		    <li>Search by text that matches a gene with only a HGNC relationship: Sln.  Expect: Sln row to match Public -- link for SLN to the HGNC homology page.</li>
    		    <li>Search by text that matches a gene with multiple orthologs; C4a or C4b.  Expect:  Either search returns the same results for the C4a,C4b row.</li>
    		    <li>Search by text that matches mouse genes without orthologs; Mug*.  Expect: 1 row for Mug2, Mug1.</li>
    		    <li>Search by text that matches a gene when HGNC trumps H-gene; Acan.  Expect: 1 row for Acan, ACAN. Human gene links to HGNC homology page.</li>
    		   </ol></li>
    		   
    	    <li>Verify search by Gene Location returns the correct data on the Grid Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by mouse coordinates; Chr5:75574916-75656722 (Kit's coords).  Expect: 10 rows on Grid includes Kit.</li>
    		    <li>Search by human coordinates; Chr4:54657928-54740715 (KIT's coords).  Expect: 1 row (Kit/KIT) on grid.</li>
    		    <li>Search by human coordinates that return multiple rows on grid; Chr4:54657928-55740715.  Expect: 6 rows on grid includes KIT.  </li>
    		   </ol></li>
    		   
    	    <li>Verify search by Gene Location returns the correct data on the Gene Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by mouse coordinates; Chr7:79053483-79115099 (Acan's coords).  Expect: 15 rows; ACAN only human gene. </li>
    		    <li>Search by human coordinates; Chr15:88803442-88875354 (ACAN's coords).  Expect: 2 rows; Acan/ACAN.</li>
    		    <li>Search by human coordinates that return add'l genes on gene tab not on grid; Chr15:88803442-88975354.  Expect: 7 genes; 3 mouse; 4 human.</li>
    		   </ol></li>
    		 
    		<li>Verify search by Gene Location returns the correct data on the Disease Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by mouse coordinates; ChrMT:13552-14070 (mt-Nd6's coords).  Expect: 2 diseases. </li>
    		    <li>Search by human coordinates; Chr11:89177858-89295759 (TYR's coordinates).  Expect: 4 diseases (includes broken out cluster). </li>
    		    <li>Search by human coordintes that match more than 1 gene; Chr11:89177858-90295759.  Expect: 5 diseases.</li>
    		   </ol></li>
    		   
    		<li>Verify search by Gene Location recognizes correct homology relationship for Grid and Gene Tabs.
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by mouse coordinates; Chr13:100124852-100137690 (Smn1's coordinates).  Expect: Smn1/SMN1,SMN2 on grid; those 3 on gene tab plus 10 other mouse genes; 4 diseases on disease tab</li>
    		    <li>Search by human coordinates; Chr5:70924941-70953015 (SMN1's coordinates).  Expect: 1 row on grid, 3 genes (Smn1, SMN1, other human); 4 diseases.</li>
    		    <li>Search by human coordinates; Chr5:70049523-70077595 (SMN2's coordinates).  Expect: 1 row on grid, 4 genes (Smn1, SMN2, 2 other human); 2 diseases.</li>
    		    <li>Search by mouse coordinates; Chr18:20376729-20410196 (Dsg1b's coordinates).  Expect: 3 rows on grid, 2 genes of homology group and 3 others on Gene tab; 2 diseases.</li>
    		    <li>Search by human coordinates; Chr18:31318089-31359246 (DSG1's coords).  Expect: 1 row on grid, 5 genes: 3 mouse orthologs, DSG1, 1 other human; 2 diseases.</li>
    		   </ol></li>
    		   
    		<li>Verify search by Phenotype (MP) returns the correct data on the Grid Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single MP id; MP:0008480 (absent eye pigmentation).  Expect: Match to 9 mouse genes.  Grid=9 rows; Gene=9 genes; Disease=9 (note that not all diseases for genes returned are on the disease tab due to only returning a subset of the genoclusters).</li>
    		    <li>Search by multiple MP ids with OR; mmmm, pppp.  Expect: </li>
    		    <li>Search by multiple MP ids with AND; mmmm AND pppp.  Expect:  </li>
    		   </ol></li>
    		   
    	    <li>Verify search by Phenotype (MP) returns the correct data on the Gene Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single MP id; xxxx.  Expect: </li>
    		    <li>Search by multiple MP ids with OR; mmmm, pppp.  Expect: </li>
    		    <li>Search by multiple MP ids with AND; mmmm AND pppp.  Expect:  </li>
    		   </ol></li>
    		 
    		<li>Verify search by Phenotype (MP) returns the correct data on the Disease Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single MP id; xxxx.  Expect: </li>
    		    <li>Search by multiple MP ids with OR; mmmm, pppp.  Expect: </li>
    		    <li>Search by multiple MP ids with AND; mmmm AND pppp.  Expect:  </li>
    		   </ol></li>
    		   
    		<li>Verify search by Disease (OMIM) returns the correct data on the Grid Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single OMIM id; xxxx.  Expect: </li>
    		    <li>Search by multiple OMIM ids with OR; mmmm, pppp.  Expect: </li>
    		    <li>Search by multiple OMIM ids with AND; mmmm AND pppp.  Expect:  </li>
    		   </ol></li>
    		   
    	    <li>Verify search by Disease (OMIM) returns the correct data on the Gene Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single OMIM id; xxxx.  Expect: </li>
    		    <li>Search by multiple OMIM ids with OR; mmmm, pppp.  Expect: </li>
    		    <li>Search by multiple OMIM ids with AND; mmmm AND pppp.  Expect:  </li>
    		   </ol></li>
    		 
    		<li>Verify search by Disease (OMIM) returns the correct data on the Disease Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single OMIM id; xxxx.  Expect: </li>
    		    <li>Search by multiple OMIM ids with OR; mmmm, pppp.  Expect: </li>
    		    <li>Search by multiple OMIM ids with AND; mmmm AND pppp.  Expect:  </li>
    		   </ol></li>
    		   
    		<li>Verify search that matches both MP and OMIM terms returns the correct data on the Grid Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single text token; xxxx.  Expect:</li>
    		    <li>Search by multiple text tokens; xxxx xxxx (implicit OR).  Expect:</li>
    		    <li>Search by multiple text tokens with AND; xxxx AND xxxx.  Expect:</li>
    		   </ol></li>
    		   
    	    <li>Verify search that matches both MP and OMIM terms returns the correct data on the Gene Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single text token; xxxx.  Expect:</li>
    		    <li>Search by multiple text tokens; xxxx xxxx (implicit OR).  Expect:</li>
    		    <li>Search by multiple text tokens with AND; xxxx AND xxxx.  Expect:</li>
    		   </ol></li>
    		 
    		<li>Verify search that matches both MP and OMIM terms returns the correct data on the Disease Tab
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single text token; xxxx.  Expect:</li>
    		    <li>Search by multiple text tokens; xxxx xxxx (implicit OR).  Expect:</li>
    		    <li>Search by multiple text tokens with AND; xxxx AND xxxx.  Expect:</li>
    		   </ol></li>
    		
    		<li>Verify search by Phenotype/Disease recognizes correct homology relationship for Grid Tab.
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Search by single text token; xxxx.  Expect:</li>
    		    <li>Search by multiple text tokens; xxxx xxxx (implicit OR).  Expect:</li>
    		    <li>Search by multiple text tokens with AND; xxxx AND xxxx.  Expect:</li>
    		   </ol></li>
    		
		</ol>
    	    
      	    
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>