<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 76</title>
</head>

<body>

<div id="maintitle"><h1>Disease Detail - Genes section</h1>
    <ul>
        <li>User Story 76, Disease Detail - Genes section</li>
    </ul>
</div><div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
	<ol>
	   <li>Content of Disease Detail - Genes section
	   <ul><li>human gene/disease associations via OMIM (and associated HomoloGene relationships if present)</li>
	       <li>mouse gene/disease associations via curators (and associated HomoloGene relationships if present)</li>
	   </ul></li>
	   
	   <li>Display Genes and Mouse Models ribbon on the Disease Detail page
	   <ol>
           <li>only display the homologs/models table if there's at least 1 gene associated to the disease</li>
           <li>display bold text line at top of ribbon as in mockup</li>
           <li>display associated human genes even when no associated mouse homolog (this is a bug right now on Public - these are not displayed, but should be)</li>
           <li>when there are no human or mouse genes associated to a disease display a special message - use the same one as on Public now</li>	   
	   </ol></li>
	   
	   <li>Display Homology/Models table
	   <ol><li>The table has 3 sections: mouse and human, mouse, human.  Only display the section if data exists</li>
	       <li>Vertical order of the table: mouse/human; mouse; human</li>
	       <li>A row in this table represents any of the following cases:
	          <ul><li>homology class that has at least 1 member that is associated to the disease (and you display all human and mouse members)</li>
	              <li>mouse gene associated to the disease, but not in a homology class</li>
	              <li>human gene associated to the disease, but not in a homology class</li>
	          </ul>
	       <li>Display styling as shown in mockup - images; headings; text; table style</li>
	   </ol></li>
	   
	   <li>Mouse Homolog column
	   <ol><li>List mouse markers in this section that pass "the rules" with an asterisk and bold text</li>
	       <li>list the mouse markers not associated to the disease but are members of the same homology class as the associated gene. (1 homology class = 1 row)</li>
	       <li>Order the list of markers -- alpha order of markers associated to disease, remaining markers not associated to disease also in alpha order.</li>
	       <li>link each mouse gene to its Marker Detail page</li>
	       <li>When there are no mouse homologs for a human gene associated to the disease - leave the column blank for that row</li>
	   </ol></li>
	   
	   <li>Human Homolog column
	       <ol><li>list the human markers associated to the disease (via OMIM association) with an asterisk and bold text per row as defined above</li>
	           <li>list the human markers not associated to the disease but are members of the same homology class as the associated gene. (1 class = 1 row)</li>
	           <li>Order the list of markers -- alpha order of markers associated to disease, remaining markers not associated to disease also in alpha order.</li>
	           <li>The human symbols are not linked anywhere (just text)</li>
	           <li>When there are no human homologs for a mouse gene associated to the disease - leave the column blank for that row</li>
	       </ol></li>
	       
	   <li>Mouse : Human Homology Class column
	       <ol><li>list the counts of mouse genes and human genes for the applicable homology class.</li>
	           <li>link the counts to the new Homology Class page</li>
	           <li>leave blank if there is no homology class associated to the human or mouse marker in that row</li>
	       </ol></li>
	       
	   <li>Criteria for including a Mouse Gene in the "Genes" section of the Disease Detail page
	       <ol>
	          <li>Mouse marker must have a Marker Type = Gene</li>
	          <li>Mouse marker must not be associated via an allele that is has a Driver Note (Cre)</li>
	          <li>Mouse marker must not be associated via a genotype that has a NOT qualifier</li>
	          <li>Mouse marker must not be associated via a Complex genotype that is NOT also conditional</li>
	          <li>Mouse marker must not be associated via a wild type allele</li>
	          <li>Mouse marker must not be associated via a Genotype that includes Gt(ROSA)26Sor
	       </ol></li>
	    
	    
	    
	    
	</ol>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
    		
    		<li>Verify Genes/Models ribbon displays 
    		<p concordion:assertTrue="true">PASS - jlewis 2/15/2013</p>  
    		<ol>
    			<li>Disease associated to no genes - OMIM id 604324; display special message</li>
    			<li>Disease associated to both human and mouse genes - OMIM id 104300; human and mouse genes w/ asterisks; display 26 genes
    			<ul><li>mouse Sorl dropped due to cx geno (moved from both to human only)</li>
    			    <li>11 mouse genes dropped from the mouse only section due to cx genotypes</li>
    			    <li>5 human genes added to the human only section - bug fix</li>
    			</ul></li>
    			<li>Disease associated to only mouse genes - OMIM id 182250; only mouse genes w/ asterisks; display 1 gene</li>
    			<li>Disease associated to only human genes - OMIM id 308930; only human genes w/ asterisks; display 1 gene</li>
    			<li>Disease associated to only human genes that have no mouse orthologs - OMIM id 609402; only human genes w/ asterisks and no mouse genes; display 1 gene</li>
       		</ol></li>
       		
       		<li>Verify vertical sections in the table display as expected
    		<p concordion:assertTrue="false">FAIL - jlewis 2/15/13; test 2.6 & 2.7 failed - mouse genes missing</p>  
    		<ol>
    			<li>Disease with only a "both" section - OMIM id 148190; display just this section; 1 gene</li>
    			<li>Disease with only a "mouse" section - OMIM id 530000; display just this section; 1 gene</li>
    			<li>Disease with only a "human" section - OMIM id 243400; display just this section; 1 gene</li>
    			<li>Disease with a "both" and "mouse" section - OMIM id 245200; display these 2 sections; 2 genes</li>
    			<li>Disease with a "both" and "human" section - OMIM id 214800; display these 2 sections; 2 genes</li>
    			<li><b>Disease with a "human" and "mouse" section - OMIM id 213300; display these 2 sections; 3 genes</b></li>
    			<li><b>Disease with a "both", "human", and "mouse section - OMIM id 604219; display these 3 sections</b>
    			<ul><li>both = 2 genes; mouse = 18 genes; human = 1 gene</li></ul></li>
    			
       		</ol></li>
       		
       		<li>Verify homology relationships are displayed correctly
    		<p concordion:assertTrue="false">FAIL - jlewis 2/15/13; test 3.6 - missing all 4 mouse genes expected</p>  
    		<ol>
    			<li>Disease with only 1-1 homology classes - OMIM id 188550; 7 human</li>
    			<li>Disease with 1-m homology class - OMIM id  ?????</li>
    			<li>Disease with n-1 homology class - OMIM id 603903; 1 mouse; 1 human</li>
    			<li>Disease with n-m homology class - OMIM id 152700; 3 both; 20 mouse; 13 human
    			<ul><li>C4a, C4b moved to "both"; 8 mouse removed due to cx geno; 2 mouse (Cd19, Lyz2) removed due to cre</li></ul></li>
    			<li>Disease with n-0 homology class - OMIM 166710; 1 both; 3 mouse; 8 human</li>
    			<li><b>Disease with 0-n homology class - OMIM 600807; 4 mouse; 13 human</b><ul><li>Human MUC7 - has 0:1 homology class</li></ul></li> 
    			<li>Disease with more than 1 gene associated from a class - OMIM id ????</li>
    			<li>Disease with a mouse gene that has no homology class - OMIM id 115650; 4 mouse; 1 human</li>
    			<li>Disease with a human gene that has no homology class - OMIM id 115650; 4 mouse; 1 human</li>
       		</ol></li>
       		
       		<li>Verify display/links/order of the 2 homolog columns
    		<p concordion:assertTrue="false">FAIL - jlewis 2/15/13; 4.2 - missing 2 mouse genes (ckr, Nrg1)</p>  
    		<ol>
    			<li>Disease with 1-1 homology classes - OMIM id 143100; 1 mouse; 1 human</li>
    			<li><b>Disease with n-m homology classes - OMIM id 181500; 2 both; 21 mouse; 24 human</b> </li>
    			<li>Disease with n-1 homology class - OMIM id 613490; 1 human (with 5 mouse homologs)</li>
       		</ol></li>
       		
       		<li>Verify "rules" have been used to filter markers included for a disease in the Genes/Models ribbon
    		<p concordion:assertTrue="false">FAIL - jlewis 2/15/13; 5.2 - missing mouse gene (cre okay); 5.6 - missing mouse gene</p>  
    		<ol>
    			<li>Disease with a marker that is not a Gene - OMIM id xxxxx</li>
    			<li><b>Disease associated to a Marker via an allele w/ Driver Note - OMIM id 142340; 7 mouse; 1 human (Nkx3-2 removed due to Cre)</b></li>
    			<li>Disease associated to a Marker via a NOT genotype annotation - OMIM id 143100; 1 both; 2 mouse (Hap1 NOT annotation)</li>
    			<li>Disease associated to a Marker via a Complex Genotype that is NOT also conditional - OMIM id 166950; no genes (due to complex genotype)</li>
    			<li>Disease associated to a Marker via a wild type allele - OMIM id ?????</li>
    			<li><b>Disease associated to a Marker via a genotype that includes Gt(ROSA) - OMIM id 114480;4 both; 5 mouse; 17 human (gt(rosa) not included)</b></li> 
       		</ol></li>
    		
		</ol>
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>