<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>US 3-US 5: SNP QF/Summary</title>
</head>

<body>

<div id="maintitle"><h1>Minimal SNP Query Form/Basic SNP Summary</h1>
    <ul>
        <li>US 3 and US 5: SNP QF and Summary - convert to fewi</li>
    </ul>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
    <ol> 
      <li>Use a minimal SNP QF with only the Associated Genes ribbon.</li>
      <li>QF/Summary display (same requirements as Public)
      <ol><li>Title</li>
          <li>user help icon/link</li>
          <li>text at top of page -- refer to SNP build (driven by database value)</li>
          <li>Remove anchor link to output options</li>
      </ol></li>
      <li>QF - Include Associated Gene ribbon
      <ol><li>Gene field - remove operators, accept wildcard, accept lists of symbols, search limted to current symbols</li>
          <li>SNPs Located Near Genes - within 2 kb of (default); other options = within coordinates, within 10 kb</li>
      </ol></li>
     <li>Use the following defaults for the query by marker symbol:
      <ol>
          <li>Only SNP variation type (No IN-DEL, Named, Mixed, or MNP variations)</li>
          <li>Include SNPs with SINGLE locations (per Carol on 12/8; don't return ones with multiple locations).  Summary may include these (original definition of story), limited testing, if any, needed.</li>
          <li>Return SNPs with dbSNP function class associations to a marker when there's a "within" search -- even if it is not "within" the coordinate range of the marker.
          <ul><li>The within coordinates search for a marker = within coordinates + dbSNP function class.</li> </ul></li>
      </ol></li>
      <li>Use new roll-up/down QF and Summary page paradigm</li>
      <li>Summary pages use the standard pagination paradigm.  Propose: use Marker Summary options: (50, 100, 250, 500) with the default = 100.</li>
      <li>Sorting
      <ol><li>Report sort order for SNPs returned: primary=by mapping location (chr/coordinate)</li>
          <li>Order for list of genes in Gene column by SNP: primary=alpha by marker symbol; secondary=alpha by dbSNP function class</li>
      </ol></li>
      <li>Display number of results returned.</li>
      <li>Display the following columns in this order (see Public)
      <ol><li>SNP ID (dbSNP Build nnn) - rs id; links to MPD, dbSNP, MGI SNP Detail (external links display new tab)</li>
          <li>Map Position (GRCm38) - chromosome; coordinate</li>
          <li>rs orient: NO LONGER DISPLAY</li>
          <li>Associated Genes - list of genes with distance association OR dbSNP Function Classes; marker symbol linked to Marker Detail</li>
          <li>Assays (ss): NO LONGER DISPLAY</li>
          <li>Variation Type - SNP</li>
          <li>Allele Summary (all strains)</li>
      </ol></li>
      <li>For this user story -- display all relationships we have for each SNP.  This includes distance, within, and dbSNP function class associations.  Refinement of this list in future user stories.  
      <ul><li>The only exception: a marker may have multiple associations to the same dbSNP function class for a gene (e.g. Intron) due to multiple RefSeqs -- only show 1.</li></ul> </li>
      <li>The SNP QF/Summary works in all supported OS/browser combinations.</li>
      <li>Do NOT include any out-of-sync logic, minimal error trapping (at SE discretion), filters, You searched for, QF striping logic.</li>
      
      
    </ol>   
   
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
		
		    <li>Query for SNPs by Marker Symbol (matching to current symbols only)
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query by text that matches a current symbol; Kit.  Expect: 1 marker matched; nn rows returned</li>
    		  <li>Query by text + wildcard that matches current symbols; Kit*.  Expect: 2 markers matched (Kit, Kitl); nn rows returned.</li>
    		  <li>Query by text + wildcard that matches current symbols; Pax6*.  Expect: 2 markers matched (Pax6, Pax6os1); nn rows returned.</li>
    		  <li>Query by text that doesn't match any markers; football.  Expect: no rows returned.</li>
              <li>Query by list of valid markers; Kit, Pax6, Lep.  Expect: 3 markers returned; nn rows returned.  Try again with Kit* (4 markers returned).</li>
              <li>Query by list of markers including an invalid one; Kit, football, Lep.  Expect:  2 markers returned; 1 error; nn rows returned.</li>
              <li>Query by marker synonym that matches no current symbols; Sox10m1 (synonym for Ednrb).  Expect: 0 markers returned; no rows returned.</li>
    		</ol></li>
    		
    		<li>Verify SNPs returned for each marker type with coordinates except QTL <b>(for this user story -- only SNPs with dbSNP function class are returned)</b>
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query by Gene marker; Tomt.  Expect: 1 marker matched; nn rows returned.</li>
    		  <li>Query by Psuedogene marker; Cnn2-ps.  Expect:  1 marker returned (only distance relationships); nn rows returned.</li>
    		  <li>Query using wildcard and special character; Rpl31-ps*.  Expect 7 markers matched (Rpl31-ps24, Rpl31-ps21, Rpl31-ps23, Rpl31-ps2, Rpl31-ps22, Rpl31-ps25, Rpl31-ps20); nn rows returned.</li>   
    		  <li>Query by Transgene marker; Tg(KITLG)3Ygy.  Expect: 0 SNPs returned (no coordinates for this marker).</li>
    		  <li>Query by Complex/Cluster/Region marker; Mirc17.  Expect:  1 marker matched (only distance relationships); nn rows returned.</li>
    		  <li>Query by Cytogenetic marker; Del(2Hoxd1-Hoxd4)55Ddu (MGI:5318060).  Expect: 1 marker matched (only distance relationships); nn rows returned.</li>
    		  <li>Query by BAC/YAC end marker; 52S.  Expect:  0 rows returned (no coordinates for this marker type).</li>
    		  <li>Query by Other Genome Feature marker; Cpgi200.  Expect: 1 marker matched (only distance relationships); nn rows returned.</li>
    		  <li>Query by DNA segment marker; D19Dcr1.  Expect:  1 marker matched (only distance relationships); nn rows returned.</li>
    		  <li>Query by heritable phenotype marker; sgl.  Expect: 1 marker matched (only distance relationships); nn rows returned.</li>
    		  <li>Query by QTL without coordinates; Asthm1 or Asthm*.  Expect: no rows returned.</li>
    		  <li>Query by QTL with coordinates; Datd.  Expect: no rows returned.</li>
    		  <li>Query by non-QTL marker that has no coordinates; D19Pas1 or Sox19.  Expect: no rows returned.</li>
    		</ol></li>
    		
    		<li>Verify all Variation types are returned (SNP, IN-DEL, Named, Mixed, MNP)
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query for marker with only SNP variations; Brca2.  Expect: nn rows returned</li>
    		  <li>Query for marker with IN-DEL variations; Hprt or Cd22 or D8Mit71.  Expect: no IN-DELs returned</li>
    		  <li>Query for marker with Named variation; D15Mit121.  Expect: no Named returned</li>
    		  <li>Query for marker with Mixed variations; Hnf4a.  Expect: no Mixed returned (2 within coords; 1 function class)</li>
    		  <li>Query for marker with MNP variations; Ube3b.  Expect: no MNP returned.</li>
    		</ol></li>
    		
    		<li>Verify all dbSNP function classes are returned
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query for marker with a Coding-NonSynonymous SNP; Trappc8.  Expect: 6 rows returned with this class.</li>
    		  <li>Query for marker with a Coding-Synonymous SNP; Cpt1a.  Expect: 5 rows returned with this class.</li>
    		  <li>Query for marker with a Intron SNP; Kdm5d.  Expect: 37 rows returned with this class.</li>
    		  <li>Query for marker with a Locus-Region SNP; Gal.  Expect: 31 rows returned with this class.</li>
    		  <li>Query for marker with a mRNA-UTR SNP; Xk.  Expect: 21 rows returned with this class.</li>
    		  <li>Query for marker with a Noncoding-Transcript-Variant SNP; Klc2.  Expect: 7 rows returned.</li>
    		  <li>Query for marker with a Splice-Site SNP; Adamts15.  Expect:  1 row returned with this class.</li>
    		</ol></li>
    		
    		<li>Verify SNPs with multiple locations are returned 
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query for marker with only SNPs with multiple locations; Sly.  Expect: 4 rows, all with asterisks; unique SNPs=2.</li>
    		  <li>Query for marker with mix of SNPs - multiple and single locations; Zdbf2.  Expect:  91 rows, 4 multiples w/ asterisks; 89=unique SNPs.</li>
    		</ol></li>
    		
    		<li>Verify that SNPs with dbSNP function class associations but outside of the MGI coordinate range ARE returned.
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query for "within coordinates of" marker with an dbSNP annotation outside of the coordinates; Trappc8.  Expect: 26 SNPs with dbSNP annotation of Locus Region returned.</li>
    		</ol></li>
    		
    		<li>Verify that SNPs on the MT chromsome are returned correctly.
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query by genes on the MT chromosome using the default "within 2kb setting"; mt-Tf, mt-Atp6, mt-Tp.  Expect: mt-Tf (only downstream SNPs), mt-Atp6 (no SNPs), mt-Tp (downstream,w/in,upstream SNPs)</li>
    		</ol></li>
    		
    		<li>Verify pagination controls - page size, default, etc.
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Run query that returns multiple pages of results.</li>
    		  <li>Run query that returns 1 page of results.</li>
    		  <li>Run query that returns no results.</li>
    		</ol></li>
    		
    		<li>Verify roll-up/down paradigm
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Run query with marker text and default location.  Expect: parameters still in place when re-opening QF.</li>
    		  <li>Run query with marker text and non-default location.  Expect: parameters still in place when re-opening QF.</li>
    		</ol></li>
    		
    		<li>Verify sort order
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Query for results on 1 chromosome.  Expect: SNPs in coordinate order.</li>
    		  <li>Query for results on multiple chromosomes; Pax*.  Expect: SNPs in chr/coordinate order (chromosomes: 1,2,4,5,6,12,16,19).</li>
    		  <li>Query for results that include multiple location SNPs.  Expect: multiple location SNPs to be in coordinate order.</li>
    		  <li>Query for results that include non-numeric and numeric chromosomes; Bmp*.  Expect: SNPs in chr/coordinate order. (chromosomes: 1,2,3,4,5,6,9,13,14,X)</li>
    		</ol></li>
    		
    		<li>Verify subsnps counts
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Return SNP with a single subsnp; Xxx.  Expect: Assay count = 1</li>
    		  <li>Return SNP with multiple subsnps; Xxx.  Expect: Assay count = nn.</li>
    		</ol></li>
    		
    		<li>Verify Gene column alpha sort order
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Return SNP with multiple associated genes; Xxx.  Expect: genes in alpha order.  </li>
    		</ol></li>
    		
    		<li>Verify Gene column displays all dbSNP function class associations
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Return SNP with a single function class; Xxx.  Expect: function class displayed</li>
    		  <li>Return SNP with multiple function classes; Xxx.  Expect: all function class annotations displayed.</li>
    		  <li>Return SNP with only a function class of "contig_reference"; Xxx.  Expect: no display of this function class.</li>
    		  <li>Return SNP with multiple function classes with one being "contig_reference"; Xxx.  Expect: all function classes except "contig_reference" displayed.</li>
    		</ol></li>
    		
    		<li>Verify Gene column displays "within coordinates of" when no function class present
    		<p concordion:assertTrue="false">new test</p>    
    		<ol>
    		  <li>Return SNP with no function class, but is located within the marker's coordinates; mt-Tp.  Expect: distance annotations displayed - upstream, downstream, and within coordinates</li>
    		</ol></li>
    		
    		<li>Test rewritten QF and Summary on all supported browsers: PC (Firefox, Chrome); Mac(Firefox, Chrome, Safari)
    		<p concordion:assertTrue="false">new test</p>   
    		<ol>
    		  <li>Repeat selected above tests for each OS/browser combination.</li>
    		</ol></li>
    		
		</ol>
    	    
      	    
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>