<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 25</title>
</head>

<body>

<div id="maintitle"><h1>GXD QF/Summary - use new roll-up annotations</h1>
    <ul>
        <li>User Story 25 - Change GXD Auto-Complete/QF/Summary to use pre-computed gene rollup annotations</li>
    </ul>
</div><div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>



<div id="acceptance">
    <h2>Test Requirements - Autocomplete and GXD QF/Summary</h2>
	<ol>
	   <li>Change the GXD QF auto-complete gene counts for MP and OMIM terms to use the new rolled-up data (simple genotypes)
	   <ul><li>The following rules to determine whether to roll-up a OMIM or MP annotation to a marker and include it in the displayed count are as follows:
	       <ol><li>Roll-up to a gene for MP/OMIM annotations for a simple 1 marker genotype.</li>
	           <li>Roll-up to a gene for MP/OMIM annotations for a conditional genotype that has just 1 marker when recombinase, transgenic reporter, and wild type alleles are factored out.</li>
	           <li>Roll-up to a gene for MP/OMIM annotations to a complex genotype that has just 1 marker when transgenic reporters are factored out.</li>
	           <li>Do not return Gt(ROSA)26Sor as a match for any MP/OMIM annotations it may have.</li>
	           <li>Factor out MP annotations to MP:0003012; "no phenotypic analysis".</li>
	           <li>Do not return any markers for matches to normal MP or NOT OMIM annotations.</li>
	       </ol></li>
	   </ul></li>
	   
	   <li>Change the GXD QF/Summary to query the data with the "rolled-up" using the new implementation of the roll-up rules on the Production server instead of the HMDC data as currently implemented.
	       <ul><li>The results are expected to be the same for the most part.  Exceptions include: slash alleles not factored out; transgenic reporter bug (factoring out when multiple attributes exist) 
	           <li>The following rules to determine whether to roll-up a OMIM or MP annotation to a marker and search expression results for it are as follows:
	            <ol><li>Roll-up to a gene for MP/OMIM annotations for a simple 1 marker genotype.</li>
	                <li>Roll-up to a gene for MP/OMIM annotations for a conditional genotype that has just 1 marker when recombinase, transgenic reporter, and wild type alleles are factored out.</li>
	                <li>Roll-up to a gene for MP/OMIM annotations to a complex genotype that has just 1 marker when transgenic reporters are factored out.</li>
	                <li>Do not return Gt(ROSA)26Sor as a match for any MP/OMIM annotations it may have.</li>
	                <li>Factor out MP annotations to MP:0003012; "no phenotypic analysis".</li>
	                <li>Do not return any markers for matches to normal MP or NOT OMIM annotations.</li>
	            </ol></li>
	      </ul></li>
	   
	
	   
	   
	</ol>
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests: ROLL_UP RULES: Auto/complete & Search Results</h3>
		
		<ol>
    		<li>Query for MP or OMIM ids that have no annotations at all
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>  
    		<ol>
    		  <li>Query for an MP term that has no annotations; MP:0008224; "increased hippocampal commissure size".  Expect: count=0; 0 results returned.</li>
    		  <li>Query for an OMIM term that has no annotations; 604324; Acne, Adult.  Expect: count=0; 0 results returned.</li>
    		  <li>Query for an OMIM term that has annotations to human genes, but no mouse OMIM annotations; 103285; ADULT Syndrome.  Expect: count=0; 0 results returned.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to simple genotypes (naturally simple; i.e. "super-simple")
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		  <li>Query for an MP term with an annotation to a simple hm genotype; MP:0005215; "abnormal pancreatic islet morphology".  Expect: Count=232; Pax4 one of the markers returned.</li>
    		   <li>Query for an MP term with an annotation to a simple genotype for a cre allele; MP:nnnnn.  Expect: </li>
    		   <li>Query for an MP term with an annotation to a simple genotype for a transgenic reporter allele; MP:nnnnn; .  Expect: </li>
    		   <li>Query for an OMIM term with an annotation to a simple hm genotype; 211980; Lung Cancer.  Expect: count=11; Robo1 one of the markers returned.</li>
    		   <li>Query for an OMIM term with an annotation to a simple genotype for a cre allele; 119540; Cleft Palate, Isolated; CPI.  Expect: count=1; Satb2 marker returned.</li>
    		   <li>Query for an OMIM term with an annotation to a simple genotype for a transgenic reporter allele; 615538; Chromosome 22q13 Duplication Syndrome.  Expect: count=1; warning message because no GXD results.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to conditional genotypes
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a conditional genotype (OMG w/cre out); MP:0009176; "increased pancreatic alpha cell number".  Expect: count=21; Pax4 one of the markers returned.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype (Complex w/cre out); MP:0002175; "decreased brain weight".  Expect: count=87; Col1a1 not returned in the results.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (OMG w/cre out); 176807; Prostate Cancer.  Expect: count=19; Apc returned in results.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (Complex w/cre out); 180200; Retinoblastoma; RB1.  Expect: count=3; warning message because there are no GXD results for those markers.</li>
    		   </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to complex genotypes
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a complex genotype; MP:0010024; "increased total body fat amount".  Expect: count=110; Adrb1 not returned in the results.</li>
    		   <li>Query for an MP term with an annotation to a non-conditional complex genotype with cre alleles; MP:0008734; "decreased susceptibility to endotoxin shock".  Expect: count=65; Elane not returned in the results.</li>
    		   <li>Query for an OMIM term with an annotation to a complex genotype; 152700; Systemic Lupus Erythematosus; SLE.  Expect: count=31; Csf2 not returned in the results.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to slash alleles
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a genotype for a slash allele: MP:0002166; "altered tumor susceptibility".  Expect: count=961; Psme2 returned.</li>
    		   <li>Query for an OMIM term with an annotation to a genotype for a slash allele; 183600; "Split-hand/foot Malformation; SHFM1".  Expect: count=2; Dlx6 returned. </li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations via transgenic reporter genotypes
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a complex genotype (OMG w/ reporter out); MP:0011128; "increased secondary ovarian follicle number".  Expect: count=4; Foxo3 in the results.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype (OMG w/cre and reporter out); MP:0006071; "abnormal retinal progenitor cell morphology".  Expect: count=12; Atoh7 to be returned in the results.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype for a transgene w/ multiple attributes including reporter (OMG w/ cre out); MP:0000774; "decreased brain size".  Expect: count=242; Tg(EEF1A1-SHC1*)1Ravi not returned (no GXD results).</li>
    		   <li>Query for an OMIM term with an annotation to a complex genotype (OMG w/reporter out); 166950; Teratoma, Ovarian.  Expect: count=2; Foxo3 returned in results.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (OMG w/cre and reporter out); nnnnn.  Expect: marker returned. (no example of this case in the real data)</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype for a transgene w/ multiple attributes including reporter (OMG w/ cre out); 174900; Juvenile Polyposis Syndrome; JPS.  Expect: count=2; Tg(CAG-Bmpr1a*,-lacz)1Nobs not returned (no GXD results).</li>
    		</ol></li>
    	    
    	    <li>Query for MP or OMIM ids with annotations involving Gt(ROSA)26Sor 
    	    <p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>   
    		<ol>
    		  <li>Query for an MP term with an annotation to a conditional genotype (Cre=Gt(ROSA)); MP:0000333; "decreased bone marrow cell number".  Expect: count=117; Nle1 returned in the results.</li>
    		  <li>Query for an MP term with an annotation to a Gt(ROSA) conditional genotype (OMG w/ cre out); MP:0000154, "rib fusion".  Expect count=76; Gt(ROSA) not returned.</li>
    		  <li>Query for an OMIM term with an annotation to a conditional genotype (Cre=Gt(ROSA)); 614286; Myelodysplastic Syndrome; MDS.  Expect: count=9; Bap1 returned in the results.</li>
    		  <li>Query for an OMIM term with an annotation to a Gt(ROSA) conditional genotype (OMG w/ cre out); 300813; Sarcoma, Synovial.  Expect count=0.</li>
    		</ol></li>
    		
    		<li>Query for MP term that only has annotations to child terms.
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Query for MP term with annotations to a child term via a simple genotype; MP:0008220; abnormal ventral commissure morphology.  Expect: count=40; Bhlhe22 returned in the results due to an annotation to MP:0008227; absent anterior commissure.</li>
    		   <li>Query for MP term with only simple genotype annotations to child terms; MP:0009120; abnormal white fat cell lipid droplet size.  Expect: 6 genes returned: Cav1, Epg5, Kdma3a, Lclat1, Lpin1, Pck1.</li>
    		   <li>Query for MP term's synonym (same term/expected results as above): "abnormal white adipocyte lipid droplet size".
    		   <li>Query for MP term with annotations to a child term via a genotype that is not simple; MP:0008220; abnormal ventral commissure morphology.  Expect:  Map1b and Map2 not returned in the results due to complex genotype annotation to a child term, MP:0008226.</li>
    		   
    	    </ol></li>
    	    
    	    
    	    <li>Query for MP annotations with a "normal" qualifier
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Query for MP term that has normal annotations to a simple genotype for a marker; MP:0005394; taste/olfaction phenotype.  Expect:  count=135; Adcy8 not returned in results.</li>
    		   <li>Query for MP term that has normal annotations to a complex genotype for a marker; MP:0005384; cellular phenotype.  Expect: Pim1 not returned in results.</li>
    		   <li>Query for MP term that has both regular and normal annotations for a marker; MP:0005394; taste/olfaction phenotype.  Expect: Cnga4 to be returned -- has a normal annotation to this term and a regular annotation to a child term.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP term - MP:0003012; "no phenotypic analysis"
    		<p concordion:assertTrue="true">PASS-jlewis 10/6/14</p>  
    		<ol>
    		   <li>Query for MP term whose annotations have been removed from the "derived set"; MP:0003012.  Expect: count=0.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Query for OMIM annotations with a "NOT" qualifier
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>  
    		<ol>
    		   <li>Query for OMIM term that has only NOT annotations to simple genotypes for a marker; 143100; Huntington Disease; HD.  Expect: count=25; Hap1 to not be returned.</li>
    		   <li>Query for OMIM term that has both regular and NOT annotations to simple genotypes for a marker; 600072; Fatal Familial Insomnia; FFI.  Expect: count=1; Prnp to be returned in the results.</li>
    	    </ol></li>
    	    
    	    <li>Query for markers using MP synonyms
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/14</p>  
    		<ol>
    		   <li>Query for an MP term using a synonym; "abnormal islet of Langerhans".  Expect: count=232.</li>
    	    </ol></li>
    	    
    	    <li>Query for markers using OMIM synonyms
    		<p concordion:assertTrue="true">PASS - jlewis 10/6/2014</p>  
    		<ol>
    		   <li>Query for an OMIM term using a synonym; "Vlcad Deficiency".  Expect: count=2; results=markers: Acadl, Acadvl </li>
    	    </ol></li>
    	    

		</ol>
	
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>