<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 120</title>
</head>

<body>

<div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>

<div id="maintitle"><h1>Load B38 gene models & associated data in PUBLIC WI</h1>
    <ul>
        <li>User Story 120, Load B38 gene models, coordinates, transcripts, proteins, raw biotypes, new markers, marker associations in PUBLIC WI.</li>
    </ul>
</div>

<div id="acceptance">
    <h2>Test Requirements</h2>
	<ul>
		<li>Verify new markers are created for NCBI, VEGA, and Ensembl.</li>    
	    <li>Verify Gene Models & their coordinates are loaded for NCBI, VEGA, and Ensembl for existing and new markers.</li>
	    <li>Verify transcripts and proteins are loaded for the Gene Models</li>
	    <li>Verify raw biotypes are loaded and translated properly for the Gene Models</li>
	    <li>Verify evidence links display and link to the appropriate page</li>
	    <li>Verify Build 38 is displayed in front-end products (all WIs, MouseBlast)</li>
	    <li>Exclude - links to external genome browsers & gbrowse are not part of this story</li>
	    <li>Verify all Build 37 gene model data is gone.
		    <ul><li>Gene models, coordinates, biotypes, transcripts, proteins, marker associations</li>
		        <li>deletion of directly assigned coordinates is NOT part of this story</li>
		        <li>deletion of SNP and GT associations is NOT part of this story</li>
		        <li>other Build 37 data/associations is NOT part of this story</li>
		    </ul>
	</ul>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		<ol>
    		<li>Verify new markers and their gene models are created for B38 in the database  
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/2012 </p> 
    		<ol>
    			<li>Query for marker <b>Gm21712</b>, this is a new novel pseudogene from Ensembl 
    			<ul><li>Expected results (line from Sophia's file):<b>Pseudogene	Gm21712	predicted gene, 21712	Y	In Progress	J:91388			ENSMUSG00000094070</b></li> 
    		    </ul></li>
    			
    			<li>Query for marker <b>Gm20918</b>, this is a new novel gene from NCBI 
    			<ul><li>Expected results (line from Sophia's file): <b>Gm20918	predicted gene, 20918	Y	In Progress	J:90438			100042443</b></li></ul>
    			    			
       		</ol></li>
       		
       		<li>Verify updated gene models for existing markers for Build 38 are updated in the database
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/2012 </p> 
    		<ol>
    		    <li>Query for marker <b>Brca2</b>, this is an existing gene with an updated gene model from Ensembl
    			<ul>Expected results (line from Sophia's file): <b>ENSMUSG00000041147	5	150522630	150569746	+	chr5:150522630-150569746, + strand. ENSEMBL annotation of mouse genome build 38, Gene type: protein_coding; Gene Name: Brca2</b>
    			</ul></li>    		
    		  	   		
    		
    		    <li>Query for marker <b>Trp53</b>, this is an existing gene with an updated gene model from VEGA
    			<ul>Expected results (line from Sophia's file): <b>OTTMUSG00000005982	11	69580359	69591873	+	chr11:69580359-69591873, + strand. VEGA annotation of mouse genome build 38, Gene type: _protein_coding; Gene Name: Trp53 </b>
    			</ul></li>
    			
    			<li>Query for marker <b>Sry</b>, this is an existing gene with an updated gene model from NCBI
    			<ul>Expected results (line from Sophia's file): <b>21674	Y	2662471	2663658	-	chrY:2662471-2663658, - strand. NCBI annotation of mouse genome build 38, GENE: Sry</b>
    			</ul></li>
    				
       		</ol></li>
       		
       		<li>Verify gene models deleted for Build 38 are no longer in the database
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/12 </p> 
    		<ol>
    		    <li>Use Batch Query to query for gene model <b>ENSMUSG00000072491</b>, this is an existing B37 gene model from Ensembl
    			<ul>Expected results: <b>no associated gene</b>
    			</ul></li>
    			
    			<li>Use Batch Query to query for gene model <b>OTTMUSG00000037917</b>, this is an existing B37 gene model from VEGA
    			<ul>Expected results: <b>no associated gene</b>
    			</ul></li>
    			
    			<li>Use Batch Query to query for gene model <b>100503451</b>, this is an existing B37 gene model from NCBI
    			<ul>Expected results: <b>Associated gene GM19699 is found -- but has only external links in the Sequence Map section.</b>
    			</ul></li>
    				
       		</ol></li>
       		
       		<li>Verify gene models that have associations to more than 1 marker. 
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/2012 </p> 
    		<ol>
    		    <li>Use the Batch Query and query for <b>ENSMUSG00000007440</b>; view markers returned for this <b>Ensembl</b> gene model
    			<ul>Expected results: <b>Pcdha1, Pcdha2, Pcdha4, Pcdha5, Pcdha10, Pcdha11, Pcdha12, Pcdhac2, Pcdhac1</b>
    			</ul></li>
    			
    			<li>Use the Batch Query and query for gene model <b>100503456</b>, view markers returned for this <b>NCBI</b> gene model.
    			<ul>Expected results: <b>Gm11342, Gm11345</b>
    			</ul></li>
    			
       		</ol></li>
       		
       		<li>Verify markers that are associated to more than 1 gene model.
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/12 </p> 
    		<ol>
    		    <li>Display the Marker Detail for marker <b>Qk</b>; view associations to <b>Ensembl</b> gene models. 
    			<ul>Expected results: <b>ENS00000044407, ENS00000062078</b>
    			</ul></li>
    			
    			<li>Display the Marker Detail for marker <b>Gm6541</b>; view associations to <b>VEGA</b> gene models.
    			<ul>Expected results: <b>OTT00000036694, OTT00000036405</b>
    			</ul></li>
    			
    			<li>Display the Marker Detail for marker <b>Gab3</b>; view associations to <b>NCBI</b> gene models
    			<ul>Expected results: <b>100503280</b> (for Build 37 an add'l model was listed here - 210710)
    			</ul></li>
    			
       		</ol></li>
       		
    		<li>Verify "Evidence" links are displayed on Marker Detail page and the link goes to Build 38 data 
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/12 (note NCBI URL broken on Public too - create TR)</p> 
    		<ol>
    			<li>Query for marker <b>Pax6</b>, this marker has evidence for all 3 providers: Ensembl, VEGA, and NCBI
    			<ul>Expected results: <b>Links in the "Other database links" section of the Marker Detail page for each provider.  Link works and goes to B38 evidence.</b>
    			</ul></li>
    			
       		</ol></li>
       		
       		<li>Verify biotype mismatch identified correctly and displayed 
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/12 </p> 
    		<ol>
    			<li>Query for marker <b>Tead1</b>; Display Biotype conflict text <i>VEGA biotypes not displayed</i>
    			<ul>Expected results: <b>conflict with Ensembl and VEGA</b>
    			</ul></li>
    						
       		</ol></li>
       	
    		
    		<li>Verify associated Build 38 Proteins/Transcripts are loaded and associated to the appropriate marker for Ensembl & VEGA
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/12</p> 
    		<ol>
    			<li>Query for marker <b>Kit</b>; Display Sequence Summary for Kit;
    			<ul>Expected results: <b>Proteins and Transcripts for VEGA and Ensembl included in the list</b>
    			</ul></li>
    			
       		</ol></li>
    		
    		<li>Verify cM positions unchanged from B37 calculations
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/12 </p> 
    		<ol>
    		    <li>Query for marker <b>Lepr</b> (a marker that existed in B37)
    			<ul>Expected results:  <b>46.96 cM</b>
    			</ul></li>
    			
    			<li>Query for marker <b>Gm21711</b> (a new marker in B38) 
    			<ul>Expected results: <b>Syntenic</b>
    			</ul></li>
    			
    			<li>Query for marker <b>Avp-rs1</b> (a marker with UN chromosome)
    			<ul>Expected results: <b>nothing displayed for cM</b>
    			</ul></li>			
    		
    			<li>Query for marker <b>DXYCbl3</b>, <b>mt-Nd5</b>, <b>DYBish6</b> (markers on XY, MT, Y)
    			<ul>Expected results: <b>Syntenic</b>
    			</ul></li> 			
       		</ol></li>
    		
    		<li>Verify build text has changed to Build 38 in PUBLIC WI and MouseBlast
    		<p concordion:assertTrue="false">FAIL - jlewis 9/18/12 - GXD Summary/marker tab displaying B37 </p> 
    		<ol>
    			<li>Python WI - Marker Summary (web, tab, excel)</li>   
    			<li>MouseBlast - Query Form</li>
    			<li>Java WI - Marker, Allele, GXD Expanded Query Forms</li> 
    			<li>Fewi - Marker Detail, Batch Query, GXD Summary (Marker tab)</li>			
       		</ol></li>
       		
       		<li>Verify build text continues to display Build 37 in the following PUBLIC WI pages
    		<p concordion:assertTrue="true">PASS - jlewis 9/18/12 </p> 
    		<ol>
    			<li>Python WI - SNP Summary</li>   
    			<li>Java WI - SNP Query Form, SNP Detail, Allele Detail</li> 	
       		</ol></li>
    		
    		
		</ol>
	</div>
	<div id="sql">
	
	<h3>SQL tests</h3>
	<I>The key for the SQL tests in this User Story is to ensure that the exporter ran correctly and moved the data from DEV1_MGI..mgd_wi1 at the end of Sprint 5 into Postgres database mgi-testdb3..pub_dev.</I><P>
Raw results are here:
	<UL>
	<LI> Sprint-5 : DEV1_MGI..mgd_wi1 (Sybase) : <A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US119_SQL_Testing.html">US119_SQL_Testing.html</A></LI>
	<LI> Sprint-6 : mgi-testdb3..pub_dev (Postgres) : <A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US120_SQL_Testing.html">US120_SQL_Testing.html</A></LI>
	</UL><P>
When comparing the counts for the two databases in these two files (above), one can see that the exporter ran faithfully to reproduce the data.  Since that is the case, the general consensus for all SQL tests for this User Story is:<BR>
<B>PASS - jsb 9/19/2012</B>.<P>
Details for individual tests are linked below:
		<ol>
    	<li>Quality/Sanity check of markers associated with gene models from the different gene model providers.<p>
    	<ol><li type=A>Counts of markers and associated gene models/transcripts/proteins.
   	
    		<blockquote><font size=-1>Comparing counts of markers and associated gene models/transcripts/proteins between DEV_MGI..mgd and the public C4AM Test area (postgres: mgi-testdb3..pub_dev) tests the overall success of the gene model loads and gene model association loads.  The distribution of counts among MGI Feature Types provides a quick check that in general, the appropriate marker types were associated with gene models, and with corresponding transcripts and proteins.  All Feature Types are included, even if there are no markers of that Feature Type associated with gene models.  This assures that all markers with a Feature Type are accounted for.</font></blockquote>
 
    		<ol>
    			<li type=1>VEGA_GM_Report<br>
    			<p concordion:assertTrue="true">PASS - jsb 9/19/12</p>
    			
    			<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
    		 	&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US120_SQL_Testing.html#VEGA_GM" TARGET="_new">VEGA_GM_Report Results, SQL</A>
    			</ul><p>
    			
    			<li type=1>Ensembl _GM_Report<br>
    			<p concordion:assertTrue="true">PASS - jsb 9/19/12</p>
    			
    			<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
    		 	&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US120_SQL_Testing.html#Ensembl_GM" TARGET="_new">Ensembl_GM_Report Results, SQL</A>
</ul><p>

    			<li type=1>NCBI _GM_Report<br>
      			<p concordion:assertTrue="false">FAIL 9/9/12 rmb &nbsp;&nbsp;&nbsp;<i>No NCBI Gene Models were loaded</i></p>

        		<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
    			&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US120_SQL_Testing.html#NCBI_GM" TARGET="_new">NCBI_GM_Report Results, SQL</A>
</ul><p><p></ol>

<li type=A>Gene Model Cardinality counts
<ol>
    			<li type="1">Markers associated with multiple GM IDs (from the same Provider)<br/>
    			<blockquote><font size=-1>This test counts the number of markers with multiple GM IDS from the same Gene Model Provider (for each Provider), comparing the numbers between DEV_MGI..mgd and the C4AM Test area (DEV1_MGI..mgd_wi1).</font></blockquote></li>
    			<p concordion:assertTrue="true">PASS - jsb 9/19/12</p>
    			<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
    			&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US120_SQL_Testing.html#NCBI_GM" TARGET="_new"><A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US120_SQL_Testing.html#Markers_MultipleIDs" TARGET="_new">Markers_Multiple_GM_IDs Results, SQL</A></A>
</ul><p><p>
    		<li type=1>Gene Model IDs associated with multiple Markers<br/>
    		<blockquote><font size=-1>This test counts the number of Gene Models associated with multiple markers (for each provider), comparing the numbers between DEV_MGI..mgd and the C4AM Test area (DEV1_MGI..mgd_wi1).</font></blockquote></li>
    			<p concordion:assertTrue="true">PASS - jsb 9/19/12</p>
    			<ul><li>expected results for this test are similar to those for B37 on DEV_MGI..mgd</li>
    			&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US120_SQL_Testing.html#GMIDs_MultipleMarkers" TARGET="_new">GeneModel_IDs_MultipleMarkers Results, SQL</A>
</ul><p><p></ol>
       		
       		
       		
       		</ol>
       		</ol>

 	</div>
</div>

</body>
</html>
