<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 158</title>
</head>

<body>

<div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>

<div id="maintitle"><h1>Load Ensembl B38 gene models & associated data in Prod WI</h1>
    <ul>
        <li>User Story 158, Load Ensembl B38 gene models, coordinates, transcripts, proteins, raw biotypes, marker associations in Prod WI.</li>
    </ul>
</div>

<div id="acceptance">
    <h2>Test Requirements</h2>
	<ul>   
	    <li>Verify Ensembl Gene Models & their coordinates are loaded for existing and new markers.</li>
	    <li>Verify transcripts and proteins are loaded for the Ensembl Gene Models</li>
	    <li>Verify raw biotypes are loaded and translated properly for the Ensembl Gene Models</li>
	    <li>Exclude - links to external genome browsers & gbrowse are not part of this story</li>
	</ul>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		<ol>
    		<li>Verify new markers and their gene models are created for B38 in the database  
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Query for marker <b>Gm21903 (MGI:5434067)</b>, this is a new novel gene with annotations from Ensembl
    			<ul>Expected results: Associated to Ensembl gene model <b>ENSMUSG00000094168; chr 17</b></ul></li>
    			
       		</ol></li>
       		
       		<li>Verify updated gene models for existing markers for Build 38 are updated in the database
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		       		
    		    <li>Query for marker <b>Trp53</b>, this is an existing gene with an updated gene model from Ensembl
    			<ul>Expected results: Gene Model <b>ENSMUSG00000059552</b>; Coordinates <b>chr11:69580359-69591873, + strand</b>
    			</ul></li>
    			
    		
       		</ol></li>
       		
       		
       		<li>Verify markers that are associated to more than 1 gene model.
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    			
    			<li>Display the Marker Detail for marker <b>Adora3 (MGI:104847)</b>; view associations to <b>ENSEMBL</b> gene models.
    			<ul>Expected results: <b>ENSMUSG00000074344, ENSMUSG00000000562</b>
    			</ul></li>
    			
       		</ol></li>
       		
       		<li>Verify gene models that are associated to more than 1 marker.
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    			
    			<li>Use the Batch Query and enter Ensembl gene model <b>ENSMUSG00000020014</b>; view associations to <b>Ensembl</b> gene models.
    			<ul>Expected results: <b>Gm872 and 4930485B16Rik</b>
    			</ul></li>
    			
       		</ol></li>
       		
       		<li>Verify biotype mismatch identified correctly and displayed 
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    			<li>Query for marker <b>Tead1</b>; Display Biotype conflict text
    			<ul>Expected results: <b>conflict with MGI and Ensembl</b>
    			</ul></li>
    						
       		</ol></li>
       		
    		
    		
    		<li>Verify associated Build 38 Proteins/Transcripts are loaded and associated to the appropriate marker for Ensembl
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    			<li>Query for marker <b>Kit</b>; Display Sequence Summary for Kit;
    			<ul>Expected results: <b>Proteins and Transcripts for Ensembl included in the list</b>
    			</ul></li>
    			
       		</ol></li>
    		
    		<li>Verify 32,548 gene model associations loaded for Build 38.
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    			<li>SQL query for Ensembl gene models</li>
    			
       		</ol></li>
    		
    		
		</ol>
	</div>
	<div id="sql">
	
	<h3>SQL tests</h3>
		SQL testing for the Ensembl gene models (US 158) is included here : <A HREF="http://prodwww.informatics.jax.org/all/wts_projects/7100/7106/US160_SQL_Testing.html" TARGET="_blank">US160_SQL_Testing</A><p>
		
       		

 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>