<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 16</title>
</head>

<body>

<div id="maintitle"><h1>Marker QF/Summary - use new roll-up annotations</h1>
    <ul>
        <li>User Story 16 - Change Marker QF/Summary to use pre-computed gene rollup annotations</li>
    </ul>
</div><div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
	<ol>
	  
	   <li>Change the Marker QF/Summary to query the data that was "rolled-up" using the new implementation of the roll-up rules on the Production server instead of the HMDC data as currently implemented.
	       <ul><li>The results are expected to be the same for the most part.  Exceptions include: slash alleles not factored out; transgenic reporter bug (factoring out when multiple attributes exist)</ul> 
	   <li>The rules to determine whether to roll-up a OMIM or MP annotation to a marker and return it for a query are as follows.   The rules have been re-implemented so need to test all cases.
	       <ol><li>Roll-up to a gene for MP/OMIM annotations for a simple 1 marker genotype.</li>
	           <li>Roll-up to a gene for MP/OMIM annotations for a conditional genotype that has just 1 marker when recombinase, transgenic reporter, and wild type alleles are factored out.</li>
	           <li>Roll-up to a gene for MP/OMIM annotations to a complex genotype that has just 1 marker when transgenic reporters are factored out.</li>
	           <li>Do not return Gt(ROSA)26Sor as a match for any MP/OMIM annotations it may have.</li>
	           <li>Factor out MP annotations to MP:0003012; "no phenotypic analysis".</li>
	           <li>Do not return any markers for matches to normal MP or NOT OMIM annotations.</li>
	       </ol></li>
	   <li>Booleans, quotes, wildcards, etc (all UI features) should continue to work as before.</li>
	   <li>Rules for searching notes has not changed -- search QTL notes, general notes, and annotation notes (except those with a normal qualifier).
	      <ul><li>Bug fix - currently public version returns matches to normal annotation notes.</li></ul></li>
	
	   
	   
	</ol>
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
    		<li>Query for MP or OMIM ids that have no annotations at all
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		  <li>Query for an MP term that has no annotations; MP:nnnnn.  Expect: 0 results returned.</li>
    		  <li>Query for an OMIM term that has no annotations; nnnnn.  Expect: 0 results returned.</li>
    		  <li>Query for an OMIM term that has human annotations, but no mouse annotations; nnnnn.  Expect: 0 results returned.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to simple genotypes (naturally simple ones or super simple)
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		  <li>Query for an MP term with an annotation to a simple hm genotype; MP:nnnnn.  Expect: xxxx.</li>
    		   <li>Query for an MP term with an annotation to a simple genotype for a cre allele; MP:nnnnn.  Expect: xxxxx</li>
    		   <li>Query for an MP term with an annotation to a simple genotype for a transgenic reporter allele; MP:nnnnn.  Expect: xxxxx</li>
    		   <li>Query for an OMIM term with an annotation to a simple hm genotype; nnnnn.  Expect: xxxx.</li>
    		   <li>Query for an OMIM term with an annotation to a simple genotype for a cre allele; nnnnn.  Expect: xxxxx</li>
    		   <li>Query for an OMIM term with an annotation to a simple genotype for a transgenic reporter allele; nnnnn.  Expect: xxxxx</li>
    		   <li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to conditional genotypes
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a conditional genotype (OMG w/cre out); MP:nnnnn.  Expect: marker returned.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype (Complex w/cre out); MP:nnnnn.  Expect: marker not returned.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (OMG w/cre out); MP:nnnnn.  Expect: marker returned.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (Complex w/cre out); MP:nnnnn.  Expect: marker not returned.</li>
    		   </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to complex genotypes
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a complex genotype; MP:nnnnn.  Expect: Xxx not returned.</li>
    		   <li>Query for an OMIM term with an annotation to a complex genotype; nnnnn.  Expect: Xxx not returned.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to slash alleles
    		<p concordion:assertTrue="false">new test</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a genotype for a slash allele: MP:0002166.  Expect: Psme1 returned.</li>
    		   <li>Query for an OMIM term with an annotation to a genotype for a slash allele; 183600.  Expect: Dlx6 returned. </li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations via transgenic reporter genotypes
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    		   <li>Query for an MP term with an annotation to a conditional genotype (OMG w/cre & reporter out); MP:nnnnn.  Expect: marker returned.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype for a transgene w/ multiple attributes including reporter (OMG w/ cre out); MP:nnnnn.  Expect: marker returned.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (OMG w/cre & reporter out); MP:nnnnn.  Expect: marker returned.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype for a transgene w/ multiple attributes including reporter (OMG w/ cre out); MP:nnnnn.  Expect: marker returned.</li>
    		</ol></li>
    	    
    	    <li>Query for MP or OMIM ids with annotations involving Gt(ROSA)26Sor 
    	    <p concordion:assertTrue="false">new test</p>  
    		<ol>
    		  <li>Query for an MP term with an annotation to a conditional genotype (Cre=Gt(ROSA)); MP:nnnnn.  Expect marker returned.</li>
    		  <li>Query for an MP term with an annotation to a Gt(ROSA) conditional genotype (OMG w/ cre out); MP:nnnnn.  Expect Gt(ROSA) not returned.</li>
    		  <li>Query for an OMIM term with an annotation to a conditional genotype (Cre=Gt(ROSA)); MP:nnnnn.  Expect marker returned.</li>
    		  <li>Query for an OMIM term with an annotation to a Gt(ROSA) conditional genotype (OMG w/ cre out); MP:nnnnn.  Expect Gt(ROSA) not returned.</li>
    		</ol></li>
    	    
    	    
    	    <li>Query for MP annotations with a "normal" qualifier
    		<p concordion:assertTrue="false">new test</p>   
    		<ol>
    		   <li>Query for MP term that has only normal annotations for a marker; MP:nnnnnn.  Expect:  Xxx to not be returned.</li>
    		   <li>Query for MP term that has both regular and normal annotations for a marker; MP:nnnnn.  Expect: XXX to be returned.</li>
    	    </ol></li>
    	    
    	    <li>Query for OMIM annotations with a "NOT" qualifier
    		<p concordion:assertTrue="false">new test</p>   
    		<ol>
    		   <li>Query for OMIM term that has only NOT annotations for a marker; 143100.  Expect: Hap1 to not be returned.</li>
    		   <li>Query for OMIM term that has both regular and NOT annotations for a marker; nnnnn.  Expect: Xxx to be returned.</li>
    	    </ol></li>
    		
    		<li>Query for MP annotations with "normal" annotation notes
    		<p concordion:assertTrue="false">new test</p>   
    		<ol>
    		   <li>Query for text that matches a normal note for an MP annotation that is in the simple set; "xxx xxx xxx".  Expect: marker to not be returned.</li>
    		   <li>Query for text that matches a normal note for an MP annotation that is not in the simple set; "xxx xxx xxx".  Expect: marker to not be returned.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP annotations with regular annotation notes
    		<p concordion:assertTrue="false">new test</p>   
    		<ol>
    		   <li>Query for text that matches a regular note for an MP annotation that is in the simple set; "xxx xxx xxx".  Expect: marker to be returned.</li>
    		   <li>Query for text that matches a regular note for an MP annotation that is not in the simple set; "xxx xxx xxx".  Expect: marker to not be returned.</li>
    	    </ol></li>
    	    
    	    
  
		</ol>
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>