<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 16</title>
</head>

<body>

<div id="maintitle"><h1>Marker QF/Summary - use new roll-up annotations</h1>
    <ul>
        <li>User Story 16 - Change Marker QF/Summary to use pre-computed gene rollup annotations</li>
    </ul>
</div><div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Test Requirements</a> | <a href="#manual">Manual Tests</a> | <a href="#sql">SQL Tests</a></p>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
	<ol>
	  
	   <li>Change the Marker QF/Summary to query the data that was "rolled-up" using the new implementation of the roll-up rules on the Production server instead of the HMDC data as currently implemented.
	       <ul><li>The results are expected to be the same for the most part.  Exceptions include: slash alleles not factored out; transgenic reporter bug (factoring out when multiple attributes exist)</ul> 
	   <li>The rules to determine whether to roll-up a OMIM or MP annotation to a marker and return it for a query are as follows.   The rules have been re-implemented so need to test all cases.
	       <ol><li>Roll-up to a gene for MP/OMIM annotations for a simple 1 marker genotype.</li>
	           <li>Roll-up to a gene for MP/OMIM annotations for a conditional genotype that has just 1 marker when recombinase, transgenic reporter, and wild type alleles are factored out.</li>
	           <li>Roll-up to a gene for MP/OMIM annotations to a complex genotype that has just 1 marker when transgenic reporters are factored out.</li>
	           <li>Do not return Gt(ROSA)26Sor as a match for any MP/OMIM annotations it may have.</li>
	           <li>Factor out MP annotations to MP:0003012; "no phenotypic analysis".</li>
	           <li>Do not return any markers for matches to normal MP or NOT OMIM annotations.</li>
	       </ol></li>
	   <li>Booleans, quotes, wildcards, etc (all UI features) should continue to work as before.</li>
	   <li>Rules for searching notes has not changed -- search QTL notes, general notes, and annotation notes (except those with a normal qualifier).
	      <ul><li>Bug fix - currently public version returns matches to normal annotation notes.</li></ul></li>
	
	   
	   
	</ol>
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests: ROLL_UP RULES</h3>
		
		<ol>
    		<li>Query for MP or OMIM ids that have no annotations at all
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>  
    		<ol>
    		  <li>Query for an MP term that has no annotations; MP:0008224; "increased hippocampal commissure size".  Expect: 0 results returned.</li>
    		  <li>Query for an OMIM term that has no annotations; 604324; Acne, Adult.  Expect: 0 results returned.</li>
    		  <li>Query for an OMIM term that has annotations to human genes, but no mouse OMIM annotations; 103285; ADULT Syndrome.  Expect: 0 results returned.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to simple genotypes (naturally simple; i.e. "super-simple")
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p> 
    		<ol>
    		  <li>Query for an MP term with an annotation to a simple hm genotype; MP:0005215; "abnormal pancreatic islet morphology".  Expect: Pax4 one of the markers returned.</li>
    		   <li>Query for an MP term with an annotation to a simple genotype for a cre allele; MP:0004031, "insulitis".  Expect: Tg(Ins2-cre)5Lt</li>
    		   <li>Query for an MP term with an annotation to a simple genotype for a transgenic reporter allele; MP:0004804; "decreased susceptibility to autoimmune diabetes".  Expect: Tg(Cd4-DsRed)4Lt/0 to be returned in results</li>
    		   <li>Query for an OMIM term with an annotation to a simple hm genotype; 211980; Lung Cancer.  Expect: Robo1 one of the markers returned.</li>
    		   <li>Query for an OMIM term with an annotation to a simple genotype for a cre allele; 119540; Cleft Palate, Isolated; CPI.  Expect: Satb2 one of the markers returned.</li>
    		   <li>Query for an OMIM term with an annotation to a simple genotype for a transgenic reporter allele; 615538; Chromosome 22q13 Duplication Syndrome.  Expect: Tg(Shank3-EGFP)1Hzo/0 to be returned in the results.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to conditional genotypes
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>
    		<ol>
    		   <li>Query for an MP term with an annotation to a conditional genotype (OMG w/cre out); MP:0009176; "increased pancreatic alpha cell number".  Expect: Pax4 one of the markers returned.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype (Complex w/cre out); MP:0002175; "decreased brain weight".  Expect: Col1a1 not returned in the results.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (OMG w/cre out); 176807; Prostate Cancer.  Expect: Apc returned in results.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (Complex w/cre out); 180200; Retinoblastoma; RB1.  Expect: Rb1 not returned in the results.</li>
    		   </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to complex genotypes
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>  
    		<ol>
    		   <li>Query for an MP term with an annotation to a complex genotype; MP:0010024; "increased total body fat amount".  Expect: Adrb1 not returned in the results.</li>
    		   <li>Query for an MP term with an annotation to a non-conditional complex genotype with cre alleles; MP:0008734; "decreased susceptibility to endotoxin shock".  Expect: Elane not returned in the results.</li>
    		   <li>Query for an OMIM term with an annotation to a complex genotype; 152700; Systemic Lupus Erythematosus; SLE.  Expect: Csf2 not returned in the results.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations to slash alleles
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>
    		<ol>
    		   <li>Query for an MP term with an annotation to a genotype for a slash allele: MP:0002166.  Expect: Psme1 returned.</li>
    		   <li>Query for an OMIM term with an annotation to a genotype for a slash allele; 183600.  Expect: Dlx6 returned. </li>
    	    </ol></li>
    	    
    	    <li>Query for MP or OMIM ids that have annotations via transgenic reporter genotypes
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p> 
    		<ol>
    		   <li>Query for an MP term with an annotation to a complex genotype (OMG w/ reporter out); MP:0011128; "increased secondary ovarian follicle number".  Expect: Foxo3 in the results.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype (OMG w/cre & reporter out); MP:0006071; "abnormal retinal progenitor cell morphology".  Expect: Atoh7 to be returned in the results.</li>
    		   <li>Query for an MP term with an annotation to a conditional genotype for a transgene w/ multiple attributes including reporter (OMG w/ cre out); MP:0000774; "decreased brain size".  Expect: Tg(EEF1A1-SHC1*)1Ravi returned in results.</li>
    		   <li>Query for an OMIM term with an annotation to a complex genotype (OMG w/reporter out); 166950; Teratoma, Ovarian.  Expect: Foxo3 returned in results.</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype (OMG w/cre & reporter out); nnnnn.  Expect: marker returned. (no example of this case in the real data)</li>
    		   <li>Query for an OMIM term with an annotation to a conditional genotype for a transgene w/ multiple attributes including reporter (OMG w/ cre out); 174900; Juvenile Polyposis Syndrome; JPS.  Expect: Tg(CAG-Bmpr1a*,-lacz)1Nobs returned in results.</li>
    		</ol></li>
    	    
    	    <li>Query for MP or OMIM ids with annotations involving Gt(ROSA)26Sor 
    	    <p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>  
    		<ol>
    		  <li>Query for an MP term with an annotation to a conditional genotype (Cre=Gt(ROSA)); MP:0000333; "decreased bone marrow cell number".  Expect: Nle1 returned in the results.</li>
    		  <li>Query for an MP term with an annotation to a Gt(ROSA) conditional genotype (OMG w/ cre out); MP:0000154, "rib fusion".  Expect Gt(ROSA) not returned.</li>
    		  <li>Query for an OMIM term with an annotation to a conditional genotype (Cre=Gt(ROSA)); 614286; Myelodysplastic Syndrome; MDS.  Expect: Bap1 returned in the results.</li>
    		  <li>Query for an OMIM term with an annotation to a Gt(ROSA) conditional genotype (OMG w/ cre out); 300813; Sarcoma, Synovial.  Expect Gt(ROSA) not returned.</li>
    		</ol></li>
    		
    		<li>Query for MP term that only has annotations to child terms.
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>   
    		<ol>
    		   <li>Query for MP term with annotations to a child term via a simple genotype; MP:0008220; abnormal ventral commissure morphology.  Expect: Bhlhe22 returned in the results due to an annotation to MP:0008227; absent anterior commissure.</li>
    		   <li>Query for MP term with only simple genotype annotations to child terms; MP:0009120; abnormal white fat cell lipid droplet size.  Expect: 6 genes returned: Cav1, Epg5, Kdma3a, Lclat1, Lpin1, Pck1.</li>
    		   <li>Query for MP term's synonym (same term/expected results as above): "abnormal white adipocyte lipid droplet size".
    		   <li>Query for MP term with annotations to a child term via a genotype that is not simple; MP:0008220; abnormal ventral commissure morphology.  Expect:  Map1b & Map2 not returned in the results due to complex genotype annotation to a child term, MP:0008226.</li>
    		   
    	    </ol></li>
    	    
    	    
    	    <li>Query for MP annotations with a "normal" qualifier
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p> 
    		<ol>
    		   <li>Query for MP term that has normal annotations to a simple genotype for a marker; MP:0005394; taste/olfaction phenotype.  Expect:  Adcy8 not returned in results.</li>
    		   <li>Query for MP term that has normal annotations to a complex genotype for a marker; MP:0005384; cellular phenotype.  Expect: Pim1 not returned in results.</li>
    		   <li>Query for MP term that has both regular and normal annotations for a marker; MP:0005394; taste/olfaction phenotype.  Expect: Cnga4 to be returned -- has a normal annotation to this term and a regular annotation to a child term.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP term - MP:0003012; "no phenotypic analysis"
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>
    		<ol>
    		   <li>Query for MP term whose annotations have been removed from the "derived set"; MP:0003012.  Expect: No results.</li>
    		   
    	    </ol></li>
    	    
    	    <li>Query for OMIM annotations with a "NOT" qualifier
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/14</p>
    		<ol>
    		   <li>Query for OMIM term that has only NOT annotations to simple genotypes for a marker; 143100; Huntington Disease; HD.  Expect: Hap1 to not be returned.</li>
    		   <li>Query for OMIM term that has both regular and NOT annotations to simple genotypes for a marker; 600072; Fatal Familial Insomnia; FFI.  Expect: Prnp to be returned in the results.</li>
    	    </ol></li>
    		
    		<li>Query for MP annotations with "normal" annotation notes
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/2014</p>   
    		<ol>
    		   <li>Query for text that matches a normal note for an MP annotation that is in the simple set; "notably, supramammillary nuclei".  Expect: marker Foxb1 to not be returned (normal note for MP:0004162; hm1 genotype for &lt;tm1Pgr&gt; allele).</li>
    		   <li>Query for text that matches a normal note for an MP annotation that is not in the simple set; "in contrast, weights of accessory sex glands (seminal vesicles and coagulating glands) are similar to those of wild-type males".  Expect: markers Thra & Thrb to not be returned.  Note is associated to annotation for MP:0004852.</li>
    	    </ol></li>
    	    
    	    <li>Query for MP annotations with backgroud sensitivity annotation notes
    		<p concordion:assertTrue="false">FAIL - jlewis 9/5 - 13.1 failed -- nothing was returned.  Same string on Public returned expected marker.</p>   
    		<ol>
    		   <li>Query for text that matches a BS note for an MP annotation that is in the simple set; "the jejunum crypt, but not ileal crypts, are more resistant".  Expect: marker Gpx1 to be returned.</li>
    		   <li>Query for text that matches a BS note for an MP annotation that is not in the simple set; "mixed genetic background of equal parts 129/Sv".  Expect: markers Thra & Thrb to not be returned. Note is associated to MP:0004738. </li>
    	    </ol></li>

    	    <li>Query for MP annotations with general annotation notes
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/2014</p>   
    		<ol>
    		   <li>Query for text that matches a general note for an MP annotation that is in the simple set; "linoleic acid, lauric acid and stearic acid stimulation" (use quotes).  Expect: Ucp1 to be returned.</li>
    		   <li>Query for text that matches a general note for an MP annotation that is not in the simple set; "cholesterol is primarily in the LDL fraction as opposed to controls where it is predominantly in the HDL fraction".  Expect: markers (Apobec1, Fgg, Ldir) to not be returned.</li>
    	    </ol></li>
    	    
    	    <li>Query for markers using MP alternate IDs and MP synonyms
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/2014</p>   
    		<ol>
    		   <li>Query for an MP term using an alternate id; MP:0001946; "abnormal pancreatic islet morphology".  Expect: Pax4 one of the markers returned.</li>
    		   <li>Query for an MP term using a synonym; "abnormal islet of Langerhans".  Expect: same results as previous example (same term).</li>
    	    </ol></li>
    	    
    	    <li>Query for markers using OMIM alternate IDs and OMIM synonyms
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/2014</p>   
    		<ol>
    		   <li>Query for an OMIM term using an alternate id; xxxxx.  Expect: </li>
    		   <li>Query for an OMIM term using a synonym; "Vlcad Deficiency".  Expect: 2 markers: Acadl, Acadvl </li>
    	    </ol></li>
    	    
  
		</ol>
		
		<h3>Manual tests: TEXT SEARCHES</h3>
		<ol>
		<li>Query for markers via QTL notes (using Phenotype Box)
    		<p concordion:assertTrue="true">FAIL - jlewis 9/5/2014</p>   
    		<ol>
    		   <li>Query for text that matches a QTL note for a QTL marker; "lung cancer" (use quotes).  Expect: 60 markers to be returned; includes QTL Clas2.</li>
    	    </ol></li>
    	 
    	 <li>Query for markers via General allele notes (using Phenotype Box)
    		<p concordion:assertTrue="true">PASS - jlewis 9/5/2014</p>   
    		<ol>
    		   <li>Query for text that matches an allele general note for a marker; "Phenotypic Similarity to Human Syndrome: Metabolic Syndrome in mice homozygous".  Expect: 3 markers; a, Apoe, Cyp19a1.</li>
    	    </ol></li>   
    	    
    	    
    	    
    	    
		</ol>
		
		<h3>Manual tests: NOMENCLATURE SEARCHES</h3>
		<ol>
		<li>Query for markers by Allele Symbol & Allele Name
    		<p concordion:assertTrue="true">FAIL - jlewis 9/5/2014; query by allele symbol results in extraneous chars in Why Matched? column</p>   
    		<ol>
    		   <li>Query for allele symbol; Lepr&lt;db&gt;.  Expect: marker Lepr returned.</li>
    		   <li>Query for allele name; yellow.  Expect: marker a returned (nonagouti).</li>
    	    </ol></li>
    	 
    	    
    	      
		</ol>
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>