<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>User Story 15</title>
</head>

<body>

<div id="TOC">
	<p>Contents: <a href="#maintitle">Title</a> | <a href="#acceptance">Acceptance Criteria</a> | <a href="#manual">Manual Tests</a> | <a href="#auto">Automated Tests</a></p>
</div>

<div id="maintitle"><h1>Assign Coordinates to Markers with Gene Models</h1>
    <ul>
        <li>User Story 16, Support bulk loading to assign coordinates directly to markers that have gene models (i.e. they have coordinates assigned via the Gene Model/Representative Sequence process).</li>
    </ul>
</div>

<div id="acceptance">
    <h2>Test Requirements</h2>
	<ul>
		<li>Assign coordinates to markers with Gene Models
		    <ul><li>marker types include: Gene, Pseudogene</li>
		        <li>gene model providers include: VEGA, Ensembl, NCBI</li>
		    </ul>
		<li>Load coordinates for multiple markers in one input file</li>
		<li>Allow curator to specify Genome Build to display in the Prod WI in the Curator input file</li>
		<li>Allow curator to specify Provider text  by marker to display in the Prod WI in the Curator input file</li>
		<li>Allow curator to override previously assigned coordinates in the Curator input file by marker</li>
		<li>Allow curator to remove previously assigned coordinates by marker making the Gene Model coordinates active again</li>
	</ul>
</div>



<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		<ol>
    		<li>Loaded chr, coordinates, provider text, and build text are displayed in the Sequence Map section of the Production WI Marker Detail page for markers with Gene Models.  
    		<p concordion:assertTrue="false">new test</p> 
    		<ol>
    			<li>Display  marker <b>xxx</b>.  This marker has a marker type of <b>Gene</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chrNN:nnn-nnn bp,  strand <br>(From xxx annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>xxx</b>.  This marker has a marker type of <b>Pseudogene</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chrNN:nnn-nnn bp,  strand <br>(From xxx annotation of NCBI Build XX)</b></li></ul>
    			
    			    			
       		</ol>
       		
    		<li>Verify newly assigned coordinates override previous coordinates defined by Gene Model.<ol>
    		<p concordion:assertTrue="false">new test</p> 	
    			<li>Display  marker <b>xxx</b>.  This marker has a gene model provider of <b>VEGA</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chrNN:nnn-nnn  bp,  strand <br>(From xxx annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>xxx</b>  This marker has a gene model provider of <b>Ensembl</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chrNN:nnn-nnn bp<br> strand(From xxx annotation of NCBI Build XX)</b></li></ul>
    			
    			<li>Display  marker <b>xxx</b>.  This marker has a gene model provider of <b>NCBI</b></li>
    			<ul><li>Verify the Sequence Map display: <br><b>chrNN:nnn-nnn bp<br>(From xxx annotation of NCBI Build XX)</b></li></ul>
    			
    		</ol>
    		
    		
    		<li>Verify assignment of coordinates with special cases in re: to chromosomes<ol>
    		<p concordion:assertTrue="false">new test</p> 
    		    <li>Display marker <b>xxx</b>. This marker is on the <b>MT chromosome</b>.</li>
    		    <ul><li>Verify the Sequence Map display: <br><b>chrMT:nnn-nnn bp, + strand<br>(From xxx annotation of NCBI Build XX)</b></li></ul>
    			
    		</ol>
    		
    		<li>Markers with no coordinate assignment via the load continue to display gene model coordinates and credit/build text. (as-of xxx - coordinates on Public WI for the following markers)<ol>
    		<p concordion:assertTrue="false">new test</p> 
    			<li>Display marker <b>xxx</b>. This marker has a marker type of <b>Gene</b> & provider of <b>VEGA</b>.</li>
                <li>Display marker <b>xxx</b>. This marker has a marker type of <b>Pseudogene</b> & provider of <b>NCBI</b>.</li>
                <li>Display marker <b>xxx</b>. This marker has a marker type of <b>Gene</b> & provider of <b>Ensembl</b>.</li>
               </ol>
    	
    			
    			
		</ol>
	</div>
	<div id="auto">

 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>