<html xmlns:concordion="http://www.concordion.org/2007/concordion" xmlns:cc="http://fewi.custom.commands.FewiExtensions">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<style>
	table {border-collapse: collapse; padding-bottom:5px; margin-bottom:5px;}
	table, th, td {border: 1px solid #CCCCCC;}
	td, th {padding:2px;}
	.man {background-color:#FFFFCC;}
	ul {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	ol {padding: 0px; padding-bottom:3px; padding-top:0px; margin: 0px;}
	li {padding-left:2px; padding-top:0px; padding-bottom:0px; margin-left:25px;}
	p {margin:2px 0px;}
</style>

    <title>US 8: Add marker ribbon to SNP Detail</title>
</head>

<body>

<div id="maintitle"><h1>SNP Detail - Marker ribbon</h1>
    <ul>
        <li>US8: Add Markers to SNP Detail</li>
    </ul>
</div>



<div id="acceptance">
    <h2>Test Requirements</h2>
    <ol> 
        <li>Display a ribbon of marker associations for a SNP.  It is possible for a SNP to have no associated markers.  In that case do not display the ribbon.</li> 
        <li>Use new styling: Title case for labels, links not underlined, wrapping</li>
        <li>Marker associations to display include:
        <ol><li>all markers that have a dbSNP function class for this SNP.</li>
            <li>all eligible markers where this SNP is "within coordinates".</li>
            <li>all eligible markers that are within 2KB upstream/downstream (or proximal/distal) of SNP coordinate.</li>
        </ol></li>
        <li>Eligible markers are those that have had associations determined in the back-end.  All markers with coordinates except QTL, Cytogenetic, and Heritable Phenotype. (assume this is driven by the data from back-end)</li>
        <li>Data displayed in this ribbon is consistent with the current Public display
        <ul><li>Exception: display of distance direction for the Locus-Region function class.</li></ul></li>
        <li>SNPs with multiple locations have 2 tables in this ribbon (location at top of each table).</li>
        <li>Location is displayed at the top of the table.  Location= chromosome, coordinate, genome build number, rs orientation</li>
        <li>Each marker with an association to the SNP is a row in the table -- certain fields may have multiple fields which will result in multiple sub-rows.</li>
        <li>All columns are displayed for each marker -- symbol, name, and function class will always have a value.  The others may be blank.</li>
        <li>Table - sort rows: smart alpha by marker symbol.
        <ol>
            <li>Symbol - marker symbol linked to Marker Detail page</li>
            <li>Name - marker name</li>
            <li>Multiple occurrences of Transcript ID, Protein ID, and/or dbSNP function class results in sub-rows for all fields except marker symbol/name.</li>
            <li>Multiple sub-rows may result in Transcript and/or Protein IDs listed multiple times.</li> 
            <li>Sub-row sort order: Primary: Transcript ID; Secondard: alpha by function class.</li>
            <li>Transcript - display ID, link to MGI Sequence Detail page.</li>
            <li>Protein - display ID, link to MGI Sequence Detail page.</li>
            <li>Function Class - dbSNP function class, within coordinates of, or within distance of.
            <ol><li>Multiple dbSNP function classes are possible for a marker - display all of them, no exclusions.</li>
                <li>Append distance direction to the dbSNP function class "Locus-Region" (part of original SNP requirements, but not part of most recent Public version)</li>
                <li>Display "{distance} bp {direction} of" for upstream/downstream distance association.</li>
                <li>Display "{distance} bp {direction}" for proximal/distal distance association.</li>
            </ol></li>
            <li>Allele - use same background color as Assay ribbon for each allele.</li>
            <li>Residue - display as stored in database</li>
            <li>Codon Position - display as stored in database</li>
            <li>AA Position - display as stored in database</li>
        </ol></li>  
        <li>Fix bugs on Public
        <ol><li>Display RefSeqs with prefix XM/XR (transcripts) and XP (proteins).  These are missing on Public even though Richard sees data there.</li>
            <li>Misalignment of some Transcript/Protein RefSeq pairs.</li>
        </ol></li>
    </ol>   
   
</div>


<div id="tests">
    <h2>Tests</h2>
    
	<div id="manual">
<!-- To insert manual test results, use the following:
<p concordion:assertTrue="true">testing assertTrue - displays as green</p>
<p concordion:assertTrue="false">testing assertFalse - displays as red</p>
 -->
		<h3>Manual tests</h3>
		
		<ol>
		
		    <li><b>Verify marker ribbon exists when a marker association is present</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/6/2016 (data displayed as loaded; deleted XM observation sent to SME/PO)</p>    
    		<ol><li>Display SNP with both distance and dbSNP function class markers; rs51567171.  Expect: Gm39552 - within coords; Uty - Intron</li>
    		    <li>Display SNP with only distance function class markers; rs8281485.  Expect: 9 MT markers with distance relationships; matches Public. </li>
    		    <li>Display SNP with only dbSNP function class markers; rs50829759.  Expect:  1 NM transcript/Intron pair.  7 XM transcripts - flagged as deleted in MGI.</li>
    		    <li>Display SNP with a marker distance association beyond 2 KB; rs33437897.  Expect: Gm25481 not in the list of marker associations. This marker has a distance relationship of 4767 upstream.  Gm41889, Xk markers listed </li>
    		    <li>Display SNP with a dbSNP function class for a marker that is beyond 2KB; rs6367566.  Expect: Gm33973 to be included in marker ribbon (dbSNP FC trumps distance limitation).</li>
    		    <li>Display SNP with no marker associations; rs33855201.  Expect:  no marker ribbon.</li>
    		    </ol></li>
		
		    <li><b>Verify SNPs that are within the coordinate range of ineligible markers do not have them listed in the Marker ribbon</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/6/2016</p>        
    		<ol><li>Display SNP in the range of a QTL; rs29662355.  Expect: QTL not listed (QTL Hivan1 overlaps). Note: case of valid XM (not deleted).</li>
    		    <li>Display SNP in the range of a Cyto marker; rs13486747.  Expect: Cyto marker not listed (Cyto marker Del(5Gtf2i-Limk1)1Uta overlaps).</li>
    		    <li>Display SNP in the range of a heritable pheno marker; rs16811814.  Expect: Pheno marker not listed (pheno marker ctl overlaps).</li>
    		    </ol></li>
    		
    		<li><b>Verify SNP with multiple locations</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>    
    		<ol><li>Display a multiple location SNP that displays 2 different genes; rs107905296.  Expect: 2 tables displayed; different locations; different genes.</li>
    		    <li>Display a multiple location SNP that displays the same gene; rs3022817.  Expect: 2 tables; 2 diff locations; same gene: D1Mit281.</li>
    		    </ol></li>
    		
    		<li><b>Verify SNP location display</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/6/2016</p>    
    		<ol><li>Display SNP with a reverse orientation; rs46633278.  Expect: Marker Syne1 (Scrumbob displays 18 XM transcripts)</li>
    		    <li>Display SNP with MT chromosome; rs33259749.  Expect: 10 markers associated via distance relationship.</li>
    		    </ol></li>
    		
    		<li><b>Verify SNPs with marker associations with no sub-rows</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>     
    		<ol><li>Display SNP with a single marker association/dbSNP FC; rs6367566.  Expect: Gm33973 - Locus-Region.</li>
    		    <li>Display SNP with a single marker association/distance FC; rs29601346.  Expect: Lep/no transcript/no protein/1251 bp downstream</li>
    		    <li>Display SNP with multiple markers/1 row each; no sub-rows; rs33259749.  Expect: 10 rows with no sub-rows, sorted alpha by symbol.</li>
    		    </ol></li>
    		 
    		 <li><b>Verify SNPs with marker associations with sub-rows (i.e. multiple values for some fields)</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>            
    		<ol>
    		    <li>Display SNP with multiple transcript ids; rs16789951.  Expect:  4 transcripts. Sorted by seq id - smart alpha.</li>
    		    <li>Display SNP with multiple protein ids; rs33823858.  Expect: 3 sets of transcripts/proteins.  Sort by transcript and function class  </li>
    		    
    		    </ol></li>
    		
    		<li><b>Verify sorting of sub-rows for a marker</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>         
    		<ol><li>Display SNP with multiple dbSNP function classes that are the same; rs29598424.  Expect:  Lep/2 transcripts/0 proteins/2 mRNA-UTR/sort by refseq id.</li>
    		    <li>Display SNP with lots of transcript IDs; rs237160095.  Expect: Pax2/11 transcripts/0 proteins/Intron.</li>
    		    <li>Display SNP with multiple dbSNP classes for a single marker; rs26922534.  Expect: Wrap3/4 transcripts/0 proteins/2 mRNA-UTR/1 Noncoding-T-V/1 Locus Region.</li>
    		    </ol></li>  
    		    
    		<li><b>Verify SNPs that include distance direction</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>            
    		<ol>
    		    <li>Display SNP with Locus-Region upstream/downstream; rs29598988 (upstream), rs29599403 (downstream).  Expect: Function class appended with appropriate direction.</li>
    		    <li>Display SNP with distance association with proximal/distal; rs36693433 (proximal), rs36522050 (distal).  Expect: appropriate direction appended.</li>
    		    </ol></li>
    		    
    		<li><b>Verify SNPs with XM, XR, XP RefSeqs</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>      
    		<ol><li>Display SNP with a XM transcript; rs48316449  Expect: Lep/XM/no protein/Intron</li>
    		    <li>Display SNP with a XR RefSeq; rs51567171.  Expect: 3 XR transcripts in Uty row.</li>
    		    <li>Display SNP with a XP protein; rs29392909.  Expect: Trp53: 8 subrows.  Multiple dbSNP classes per transcript - verify sort.  All rows have transcript+protein.</li>
    		    </ol></li>
    		    
    		<li><b>Verify SNPs with Alleles in marker ribbon</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>          
    		<ol><li>Display SNPs that display all types of alleles; rs33823080 (A/T); rs33823858 (A/G); rs29607066 (C).  Expect: Display to match colors in previous Assay ribbon on page.</li>
    		    <li>Display SNP with alleles of (-); rs16789951.  Expect: Display to reflect Solr data listed beneath page.SEND TO RMB FOR REVIEW.</li>
    		    </ol></li>   
    		 
    		<li><b>Verify SNPs with misalignment issues on Public now (example from rmb)</b>
    		<p concordion:assertTrue="true">PASS - jlewis; 1/7/2016</p>           
    		<ol><li>Display SNP with issue on Public; rs6314659.  Expect: transcripts and proteins to be aligned properly in table.</li>
    		</ol></li>  
    		    
    		<li>Verify assay ribbon displays okay in all supported browsers.
    		<p concordion:assertTrue="false">new test</p>    
    		<ol><li>Use selected above tests to verify.</li>
    		    </ol></li>
    		
    		
		</ol>
    	    
      	    
	</div>
	<div id="sql">
	
	
 <!--  EXAMPLE TEST CASE USING DYNAMIC DATA SPREADSHEET AT https://docs.google.com/spreadsheet/ccc?key=0ArepxibBtJW2dEl0eEg1bElOc1Mxc1FnNGtmSmlUSlE#gid=2 
     			<li>
    			Query for <b concordion:set="#structure">4-cell stage</b> should return 
    			<b cc:assertEquals="getResultCount(#structure)"><b cc:dynamicData="countFor4CellStageResults"></b></b> assay results, 
    			<b cc:assertEquals="getAssayCount(#structure)"><b cc:dynamicData="countFor4CellStageAssays"></b></b> assays, 
    			and <b cc:assertEquals="getGeneCount(#structure)"><b cc:dynamicData="countFor4CellStageGenes"></b></b> genes.<br/></li>
  -->
 	</div>
</div>

</body>
</html>